diff -up ./nss/lib/softoken/pkcs11c.c.pkcs12-rsa-pss ./nss/lib/softoken/pkcs11c.c
--- ./nss/lib/softoken/pkcs11c.c.pkcs12-rsa-pss	2017-03-23 15:32:39.172593148 -0700
+++ ./nss/lib/softoken/pkcs11c.c	2017-03-23 15:33:20.789301538 -0700
@@ -5592,6 +5592,7 @@ sftk_unwrapPrivateKey(SFTKObject *key, S
 
     switch (SECOID_GetAlgorithmTag(&pki->algorithm)) {
         case SEC_OID_PKCS1_RSA_ENCRYPTION:
+        case SEC_OID_PKCS1_RSA_PSS_SIGNATURE:
             keyTemplate = nsslowkey_RSAPrivateKeyTemplate;
             paramTemplate = NULL;
             paramDest = NULL;
