diff -up ./nss/lib/softoken/pkcs11c.c.1154764 ./nss/lib/softoken/pkcs11c.c
--- ./nss/lib/softoken/pkcs11c.c.1154764	2015-01-12 13:53:36.488856046 -0800
+++ ./nss/lib/softoken/pkcs11c.c	2015-01-12 13:53:36.552856584 -0800
@@ -3523,12 +3523,12 @@ nsc_parameter_gen(CK_KEY_TYPE key_type,
 	}
 	params->arena = arena;
 	crv = sftk_Attribute2SSecItem(arena, &params->prime, key, CKA_PRIME);
-	if (rv != SECSuccess) {
+	if (crv != CKR_OK) {
 	    goto loser;
 	}
 	crv = sftk_Attribute2SSecItem(arena, &params->subPrime, 
 							key, CKA_SUBPRIME);
-	if (crv != SECSuccess) {
+	if (crv != CKR_OK) {
 	    goto loser;
 	}
 
@@ -3544,7 +3544,7 @@ nsc_parameter_gen(CK_KEY_TYPE key_type,
 	}
 	vfy->arena = arena;
 	crv = sftk_Attribute2SSecItem(arena, &vfy->seed, key, CKA_NSS_PQG_SEED);
-	if (rv != SECSuccess) {
+	if (crv != CKR_OK) {
 	    goto loser;
 	}
 	crv = sftk_Attribute2SSecItem(arena, &vfy->h, key, CKA_NSS_PQG_H);
