diff -up ./mozilla/security/nss/lib/softoken/pkcs11.c.nocheckfork ./mozilla/security/nss/lib/softoken/pkcs11.c
--- ./mozilla/security/nss/lib/softoken/pkcs11.c.nocheckfork	2014-03-11 15:07:08.279347367 -0700
+++ ./mozilla/security/nss/lib/softoken/pkcs11.c	2014-03-11 15:08:42.773414078 -0700
@@ -2130,8 +2130,6 @@ sftk_IsWeakKey(unsigned char *key,CK_KEY
 /* return the function list */
 CK_RV NSC_GetFunctionList(CK_FUNCTION_LIST_PTR *pFunctionList)
 {
-    CHECK_FORK();
-
     *pFunctionList = (CK_FUNCTION_LIST_PTR) &sftk_funcList;
     return CKR_OK;
 }
@@ -2139,8 +2137,6 @@ CK_RV NSC_GetFunctionList(CK_FUNCTION_LI
 /* return the function list */
 CK_RV C_GetFunctionList(CK_FUNCTION_LIST_PTR *pFunctionList)
 {
-    CHECK_FORK();
-
     return NSC_GetFunctionList(pFunctionList);
 }
 
