diff --git a/yum/update_md.py b/yum/update_md.py
index 103149a9..5c91232a 100644
--- a/yum/update_md.py
+++ b/yum/update_md.py
@@ -611,8 +611,7 @@ class UpdateMetadata(object):
                     msg = _("Update notice %s%s is broken, or a bad duplicate, skipping.") % (un['update_id'], _rid(repoid))
                     if not have_dup:
                         msg += _('\nYou should report this problem to the owner of the %srepository.') % _rid(repoid, "%s ")
-                        msg += _('\nIf you are the owner, consider re-running the same command with --verbose to see the '
-                                 'exact data that caused the conflict.')
+                        msg += _('\nTo help pinpoint the issue, please attach the output of "yum updateinfo --verbose" to the report.')
                     have_dup = True
                     if self._vlogger:
                         self._vlogger.warn("%s", msg)
