From 660d3a6f150a59433f78a1ca330e446b7846d4b8 Mon Sep 17 00:00:00 2001
From: Richard Marko <rmarko@fedoraproject.org>
Date: Wed, 20 May 2015 14:36:28 +0200
Subject: [PATCH] cli: remove dead code

Related: #1224984

Signed-off-by: Richard Marko <rmarko@fedoraproject.org>
---
 src/cli/abrt-cli.c | 19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/src/cli/abrt-cli.c b/src/cli/abrt-cli.c
index bc11c7f..8e19081 100644
--- a/src/cli/abrt-cli.c
+++ b/src/cli/abrt-cli.c
@@ -75,25 +75,6 @@ static unsigned handle_internal_options(int argc, const char **argv, const char
         {
             return skip + argc;
         }
-#if 0
-        if (prefixcmp(cmd, "--base-dir=") == 0)
-            D_list = g_list_append(D_list, xstrdup(cmd + strlen("--base-dir=")));
-        else if (prefixcmp(cmd, "--list-events") == 0)
-        {
-            const char *pfx = cmd + strlen("--list-events");
-            if (pfx && *pfx)
-                pfx += 1; /* skip '=' */
-
-            char *events = list_possible_events(NULL, dump_dir_name, pfx);
-            if (!events)
-                exit(1); /* error msg is already logged */
-
-            fputs(events, stdout);
-            free(events);
-
-            exit(0);
-        }
-#endif
         else
             error_msg_and_die("%s", usage);
 
-- 
2.4.3

