From 5dcccd8af2cf36257d43450f1459175a7b579526 Mon Sep 17 00:00:00 2001
From: Jakub Filak <jfilak@redhat.com>
Date: Wed, 8 Jan 2014 17:11:20 +0100
Subject: [LIBREPORT PATCH 3/6] don't offer uploading on failure in report-gtk

Signed-off-by: Jakub Filak <jfilak@redhat.com>
---
 src/gui-wizard-gtk/wizard.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/src/gui-wizard-gtk/wizard.c b/src/gui-wizard-gtk/wizard.c
index 3faa8e8..e962c11 100644
--- a/src/gui-wizard-gtk/wizard.c
+++ b/src/gui-wizard-gtk/wizard.c
@@ -1719,14 +1719,7 @@ _("Processing of the problem failed. This can have many reasons but there are tw
 "\t▫ <b>network connection problems</b>\n"\
 "\t▫ <b>corrupted problem data</b>\n"));
 
-    add_warning(
-_("If you want to help us, please click on the upload button and provide all problem data for a deep analysis.\n"\
-"<i>Before you do that, please consider the security risks. Problem data may contain sensitive information like passwords.\n"\
-"The uploaded data are stored in a protected storage and only a limited number of persons can read them.</i>"));
-
     show_warnings();
-
-    gtk_widget_show(g_btn_onfail);
 }
 
 static gboolean consume_cmd_output(GIOChannel *source, GIOCondition condition, gpointer data)
-- 
1.8.3.1

