From aebf11fd1db8e67e6593305e5aeba1106565a613 Mon Sep 17 00:00:00 2001
From: Siteshwar Vashisht <svashisht@redhat.com>
Date: Wed, 15 Jul 2015 12:23:26 +0530
Subject: [PATCH] don't spit unnecessary debug messages

Related: #1243280

Signed-off-by: Jakub Filak <jfilak@redhat.com>
---
 src/lib/dump_dir.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/lib/dump_dir.c b/src/lib/dump_dir.c
index 796f947..0575f11 100644
--- a/src/lib/dump_dir.c
+++ b/src/lib/dump_dir.c
@@ -299,14 +299,14 @@ static const char *dd_check(struct dump_dir *dd)
     dd->dd_time = parse_time_file_at(dd->dd_fd, FILENAME_TIME);
     if (dd->dd_time < 0)
     {
-        log_warning("Missing file: "FILENAME_TIME);
+        log_debug("Missing file: "FILENAME_TIME);
         return FILENAME_TIME;
     }
 
     dd->dd_type = load_text_file_at(dd->dd_fd, FILENAME_TYPE, DD_LOAD_TEXT_RETURN_NULL_ON_FAILURE);
     if (!dd->dd_type || (strlen(dd->dd_type) == 0))
     {
-        log_warning("Missing or empty file: "FILENAME_TYPE);
+        log_debug("Missing or empty file: "FILENAME_TYPE);
         return FILENAME_TYPE;
     }
 
@@ -347,7 +347,7 @@ static int dd_lock(struct dump_dir *dd, unsigned sleep_usec, int flags)
         if (missing_file)
         {
             xunlinkat(dd->dd_fd, ".lock", /*only files*/0);
-            log_warning("Unlocked '%s' (no or corrupted '%s' file)", dd->dd_dirname, missing_file);
+            log_notice("Unlocked '%s' (no or corrupted '%s' file)", dd->dd_dirname, missing_file);
             if (--count == 0 || flags & DD_DONT_WAIT_FOR_LOCK)
             {
                 errno = EISDIR; /* "this is an ordinary dir, not dump dir" */
-- 
2.4.3

