From eac1345cca325609b3ded819b4439627ebe3cf8f Mon Sep 17 00:00:00 2001
From: Jakub Filak <jfilak@redhat.com>
Date: Thu, 17 Sep 2015 13:50:44 +0200
Subject: [PATCH] testsuite: ureport: use less strange testing error message

Signed-off-by: Jakub Filak <jfilak@redhat.com>
---
 tests/ureport.at | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/ureport.at b/tests/ureport.at
index 2e844b7..49c9652 100644
--- a/tests/ureport.at
+++ b/tests/ureport.at
@@ -707,7 +707,7 @@ int main(void)
     memset((void *)&ps, 0, sizeof(ps));
 
     ps.curl_result = 1;
-    strcpy(ps.errmsg, "err");
+    strcpy(ps.errmsg, "Artificial Error for the purpose of testing ability to recover from errors");
     ps.body = (char *)"body";
 
     struct ureport_server_config config;
-- 
2.4.3

