From e9a19a54e57766895767798da15da953078390a8 Mon Sep 17 00:00:00 2001
From: Martin Kutlak <mkutlak@redhat.com>
Date: Thu, 27 Sep 2018 13:49:54 +0200
Subject: [PATCH] testsuite: Add macros for value comparison

Signed-off-by: Martin Kutlak <mkutlak@redhat.com>
---
 tests/helpers/testsuite.h | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/tests/helpers/testsuite.h b/tests/helpers/testsuite.h
index 28bfd3eb..bc7abf1d 100644
--- a/tests/helpers/testsuite.h
+++ b/tests/helpers/testsuite.h
@@ -168,9 +168,22 @@ FILE *g_testsuite_output_stream = 0;
 #define TS_ASSERT_SIGNED_EQ(actual, expected) \
     TS_ASSERT_SIGNED_OP_MESSAGE(actual, ==, expected, NULL)
 
+
+#define TS_ASSERT_SIGNED_NEQ(actual, expected) \
+    TS_ASSERT_SIGNED_OP_MESSAGE(actual, !=, expected, NULL)
+
 #define TS_ASSERT_SIGNED_GE(actual, expected) \
     TS_ASSERT_SIGNED_OP_MESSAGE(actual, >=, expected, NULL)
 
+#define TS_ASSERT_SIGNED_GT(actual, expected) \
+    TS_ASSERT_SIGNED_OP_MESSAGE(actual, >, expected, NULL)
+
+#define TS_ASSERT_SIGNED_LE(actual, expected) \
+    TS_ASSERT_SIGNED_OP_MESSAGE(actual, <=, expected, NULL)
+
+#define TS_ASSERT_SIGNED_LT(actual, expected) \
+    TS_ASSERT_SIGNED_OP_MESSAGE(actual, <, expected, NULL)
+
 
 /*
  * Testing of chars
-- 
2.17.2

