--- a/sysdeps/arm/dl-machine.h	2018-09-28 17:03:45.251492319 +0200
+++ b/sysdeps/arm/dl-machine.h	2018-09-28 17:23:33.426088066 +0200
@@ -296,6 +296,7 @@
 /* ARM never uses Elf32_Rela relocations for the dynamic linker.
    Prelinked libraries may use Elf32_Rela though.  */
 #define ELF_MACHINE_NO_RELA defined RTLD_BOOTSTRAP
+#define ELF_MACHINE_NO_REL 0
 
 /* Names of the architecture-specific auditing callback functions.  */
 #define ARCH_LA_PLTENTER arm_gnu_pltenter
