diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -33,7 +33,6 @@
 IT_FIND_TOOL([TAR], [tar])
 IT_FIND_TOOL([CHMOD], [chmod])
 IT_FIND_TOOL([SHA256SUM], [sha256sum])
-IT_FIND_TOOL([WGET], [wget])
 IT_FIND_TOOL([ZIP], [zip])
 IT_FIND_TOOL([UNZIP], [unzip])
 IT_FIND_TOOL([CPIO], [cpio])
@@ -268,6 +267,9 @@
   IT_FIND_ECJ_JAR
   IT_FIND_TOOL([XSLTPROC], [xsltproc])
 fi
+if test "x$enable_downloading" = "xyes"; then
+   IT_FIND_TOOL([WGET], [wget])
+fi
 IT_USING_CACAO
 AC_CONFIG_FILES([javac], [chmod +x javac])
 AC_CONFIG_FILES([javap], [chmod +x javap])
