diff --git a/Lib/test/test_minidom.py b/Lib/test/test_minidom.py
index ca0f0f7..eff4f52 100644
--- a/Lib/test/test_minidom.py
+++ b/Lib/test/test_minidom.py
@@ -5,12 +5,14 @@ from StringIO import StringIO
 from test.test_support import verbose, run_unittest, findfile
 import unittest
 
+import pyexpat
 import xml.dom
 import xml.dom.minidom
 import xml.parsers.expat
 
 from xml.dom.minidom import parse, Node, Document, parseString
 from xml.dom.minidom import getDOMImplementation
+from xml.parsers.expat import ExpatError
 
 
 tstfile = findfile("test.xml", subdir="xmltestdata")
@@ -1062,8 +1064,10 @@ class MinidomTest(unittest.TestCase):
 
         # Verify that character decoding errors raise exceptions instead
         # of crashing
-        self.assertRaises(UnicodeDecodeError, parseString,
-                '<fran\xe7ais>Comment \xe7a va ? Tr\xe8s bien ?</fran\xe7ais>')
+        self.assertRaises(ExpatError, parseString,
+                '<fran\xe7ais></fran\xe7ais>')
+        self.assertRaises(ExpatError, parseString,
+                '<franais>Comment \xe7a va ? Tr\xe8s bien ?</franais>')
 
         doc.unlink()
 
diff --git a/Lib/test/test_xml_etree.py b/Lib/test/test_xml_etree.py
index 4e161ca..bf92361 100644
--- a/Lib/test/test_xml_etree.py
+++ b/Lib/test/test_xml_etree.py
@@ -1797,18 +1797,6 @@ def check_issue6233():
 
     """
 
-def check_issue3151():
-    """
-
-    >>> e = ET.XML('<prefix:localname xmlns:prefix="${stuff}"/>')
-    >>> e.tag
-    '{${stuff}}localname'
-    >>> t = ET.ElementTree(e)
-    >>> ET.tostring(e)
-    '<ns0:localname xmlns:ns0="${stuff}" />'
-
-    """
-
 def check_issue6565():
     """
 
