From c3e79f3d2186a7803cec352c10ba1bcf5d06fed6 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 23 Oct 2021 15:08:16 +0200
Subject: [PATCH] added arm to the ExclusiveArch and nobuildarches lists

---
 SPECS/kernel.spec | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/SPECS/kernel.spec b/SPECS/kernel.spec
index 8b663a2..c739cf6 100644
--- a/SPECS/kernel.spec
+++ b/SPECS/kernel.spec
@@ -229,6 +229,11 @@ Summary: The Linux kernel
 %define with_kdump 1
 %endif
 
+%ifarch %{arm}
+%define asmarch arm
+%define hdrarch arm
+%endif
+
 #cross compile make
 %if %{with_cross}
 %define cross_opts CROSS_COMPILE=%{cross_target}-linux-gnu-
@@ -248,7 +253,7 @@ Summary: The Linux kernel
 # Which is a BadThing(tm).
 
 # We only build kernel-headers on the following...
-%define nobuildarches i686 s390 ppc
+%define nobuildarches i686 s390 ppc %{arm}
 
 %ifarch %nobuildarches
 %define with_default 0
@@ -336,10 +341,10 @@ Group: System Environment/Kernel
 License: GPLv2
 URL: http://www.kernel.org/
 Version: %{rpmversion}
-Release: %{pkg_release}
+Release: %{pkg_release}.redsleeve
 # DO NOT CHANGE THE 'ExclusiveArch' LINE TO TEMPORARILY EXCLUDE AN ARCHITECTURE BUILD.
 # SET %%nobuildarches (ABOVE) INSTEAD
-ExclusiveArch: noarch i686 x86_64 ppc ppc64 ppc64le s390 s390x
+ExclusiveArch: noarch i686 x86_64 ppc ppc64 ppc64le s390 s390x %{arm}
 ExclusiveOS: Linux
 
 %kernel_reqprovconf
@@ -1813,6 +1818,9 @@ fi
 %kernel_variant_files %{with_kdump} kdump
 
 %changelog
+* Sat Oct 23 2021 Jacco Ligthart <jacco@redsleeve.org> - 3.10.0-1160.45.1.el7.redsleeve
+- added arm to the ExclusiveArch and nobuildarches lists
+
 * Fri Sep 24 2021 Rado Vrbovsky <rvrbovsk@redhat.com> [3.10.0-1160.45.1.el7]
 - CI: handle RT branches in a single config (Veronika Kabatova)
 - CI: Drop private CI config (Veronika Kabatova)
-- 
1.8.3.1

