From 6b99739598800f30b5cfca7d747715ba2a9328c7 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Tue, 30 Apr 2019 23:13:37 +0200
Subject: [PATCH] added arm to the ExclusiveArch and nobuildarches lists

---
 SPECS/kernel.spec | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/SPECS/kernel.spec b/SPECS/kernel.spec
index 52b3582..83ea012 100644
--- a/SPECS/kernel.spec
+++ b/SPECS/kernel.spec
@@ -222,6 +222,11 @@ Summary: The Linux kernel
 %define with_kdump 1
 %endif
 
+%ifarch %{arm}
+%define asmarch arm
+%define hdrarch arm
+%endif
+
 #cross compile make
 %if %{with_cross}
 %define cross_opts CROSS_COMPILE=%{cross_target}-linux-gnu-
@@ -241,7 +246,7 @@ Summary: The Linux kernel
 # Which is a BadThing(tm).
 
 # We only build kernel-headers on the following...
-%define nobuildarches i686 s390 ppc
+%define nobuildarches i686 s390 ppc %{arm}
 
 %ifarch %nobuildarches
 %define with_default 0
@@ -329,10 +334,10 @@ Group: System Environment/Kernel
 License: GPLv2
 URL: http://www.kernel.org/
 Version: %{rpmversion}
-Release: %{pkg_release}
+Release: %{pkg_release}.redsleeve
 # DO NOT CHANGE THE 'ExclusiveArch' LINE TO TEMPORARILY EXCLUDE AN ARCHITECTURE BUILD.
 # SET %%nobuildarches (ABOVE) INSTEAD
-ExclusiveArch: noarch i686 x86_64 ppc ppc64 ppc64le s390 s390x
+ExclusiveArch: noarch i686 x86_64 ppc ppc64 ppc64le s390 s390x %{arm}
 ExclusiveOS: Linux
 
 %kernel_reqprovconf
@@ -1757,6 +1762,9 @@ fi
 %kernel_variant_files %{with_kdump} kdump
 
 %changelog
+* Tue Apr 30 2019 Jacco Ligthart <jacco@redsleeve.org> - 3.10.0-957.12.1.el7.redsleeve
+- added arm to the ExclusiveArch and nobuildarches lists
+
 * Tue Apr 23 2019 CentOS Sources <bugs@centos.org> - 3.10.0-957.12.1.el7
 - Apply debranding changes
 - Sign with new Secureboot key
-- 
1.8.3.1

