From f55e2dc3d845a4a405a906b4f3fd5fd5deeeae93 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Wed, 10 Apr 2019 00:49:27 +0200
Subject: [PATCH] added arm to the ExclusiveArch and nobuildarches lists

---
 SPECS/kernel.spec | 18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

diff --git a/SPECS/kernel.spec b/SPECS/kernel.spec
index 1c77c8e..15151fc 100644
--- a/SPECS/kernel.spec
+++ b/SPECS/kernel.spec
@@ -222,6 +222,11 @@ Summary: The Linux kernel
 %define with_kdump 1
 %endif
 
+%ifarch %{arm}
+%define asmarch arm
+%define hdrarch arm
+%endif
+
 #cross compile make
 %if %{with_cross}
 %define cross_opts CROSS_COMPILE=%{cross_target}-linux-gnu-
@@ -241,7 +246,7 @@ Summary: The Linux kernel
 # Which is a BadThing(tm).
 
 # We only build kernel-headers on the following...
-%define nobuildarches i686 s390 ppc
+%define nobuildarches i686 s390 ppc %{arm}
 
 %ifarch %nobuildarches
 %define with_default 0
@@ -329,10 +334,10 @@ Group: System Environment/Kernel
 License: GPLv2
 URL: http://www.kernel.org/
 Version: %{rpmversion}
-Release: %{pkg_release}
+Release: %{pkg_release}.redsleeve
 # DO NOT CHANGE THE 'ExclusiveArch' LINE TO TEMPORARILY EXCLUDE AN ARCHITECTURE BUILD.
 # SET %%nobuildarches (ABOVE) INSTEAD
-ExclusiveArch: noarch i686 x86_64 ppc ppc64 ppc64le s390 s390x
+ExclusiveArch: noarch i686 x86_64 ppc ppc64 ppc64le s390 s390x %{arm}
 ExclusiveOS: Linux
 
 %kernel_reqprovconf
@@ -396,11 +401,11 @@ Source11: x509.genkey
 Source12: extra_certificates
 %if %{?released_kernel}
 Source13: centos-ca-secureboot.der
-Source14: centossecureboot001.crt
+#Source14: centossecureboot001.crt
 %define  centossecureboot001
 %else
 Source13: centos-ca-secureboot.der
-Source14: centossecureboot001.crt
+#Source14: centossecureboot001.crt
 %define  centossecureboot001
 %endif
 Source15: centos-ldup.x509
@@ -1757,6 +1762,9 @@ fi
 %kernel_variant_files %{with_kdump} kdump
 
 %changelog
+* Thu Mar 14 2019 Jacco Ligthart <jacco@redsleeve.org> - 3.10.0-957.10.1.el7.redsleeve
+- added arm to the ExclusiveArch and nobuildarches lists
+
 * Tue Mar 12 2019 CentOS Sources <bugs@centos.org> - 3.10.0-957.10.1.el7
 - Apply debranding changes
 - Sign with new secureboot key
-- 
1.8.3.1

