From 9c2c6873bc5fa2d1aeed1b63dbe92c37e24186b2 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Thu, 29 Jun 2017 20:55:19 +0200
Subject: [PATCH] added arm to the ExclusiveArch and nobuildarches lists

---
 SPECS/kernel.spec | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/SPECS/kernel.spec b/SPECS/kernel.spec
index d33f39c..5d8e16d 100644
--- a/SPECS/kernel.spec
+++ b/SPECS/kernel.spec
@@ -198,6 +198,11 @@ Summary: The Linux kernel
 %define with_kdump 1
 %endif
 
+%ifarch %{arm}
+%define asmarch arm
+%define hdrarch arm
+%endif
+
 #cross compile make
 %if %{with_cross}
 %define cross_opts CROSS_COMPILE=%{cross_target}-linux-gnu-
@@ -216,7 +221,7 @@ Summary: The Linux kernel
 # Which is a BadThing(tm).
 
 # We only build kernel-headers on the following...
-%define nobuildarches i686 s390 ppc
+%define nobuildarches i686 s390 ppc %{arm}
 
 %ifarch %nobuildarches
 %define with_default 0
@@ -296,10 +301,10 @@ Group: System Environment/Kernel
 License: GPLv2
 URL: http://www.kernel.org/
 Version: %{rpmversion}
-Release: %{pkg_release}
+Release: %{pkg_release}.redsleeve
 # DO NOT CHANGE THE 'ExclusiveArch' LINE TO TEMPORARILY EXCLUDE AN ARCHITECTURE BUILD.
 # SET %%nobuildarches (ABOVE) INSTEAD
-ExclusiveArch: noarch i686 x86_64 ppc ppc64 ppc64le s390 s390x
+ExclusiveArch: noarch i686 x86_64 ppc ppc64 ppc64le s390 s390x %{arm}
 ExclusiveOS: Linux
 
 %kernel_reqprovconf
@@ -1573,6 +1578,9 @@ fi
 %kernel_variant_files %{with_kdump} kdump
 
 %changelog
+* Thu Jun 29 2017 Jacco Ligthart <jacco@redsleeve.org> - 3.10.0-514.26.1.el7.redsleeve
+- added arm to the ExclusiveArch and nobuildarches lists
+
 * Wed Jun 28 2017 CentOS Sources <bugs@centos.org> - 3.10.0-514.26.1.el7
 - Apply debranding changes
 
-- 
1.8.3.1

