From 58b9c51c07fdd5f8a28b8fececc916181d671eee Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Tue, 13 Dec 2016 21:59:33 +0100
Subject: [PATCH] added "--tag=CC" to the make command due to libtool errors

---
 SPECS/pidgin.spec | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/SPECS/pidgin.spec b/SPECS/pidgin.spec
index e94ba69..865d849 100644
--- a/SPECS/pidgin.spec
+++ b/SPECS/pidgin.spec
@@ -112,7 +112,7 @@
 
 Name:           pidgin
 Version:        2.10.7
-Release:        26%{?dist}
+Release:        26%{?dist}.redsleeve
 License:        GPLv2+ and GPLv2 and MIT
 # GPLv2+ - libpurple, gnt, finch, pidgin, most prpls
 # GPLv2 - silc & novell prpls
@@ -579,11 +579,11 @@ autoreconf --force --install
            --enable-tcl --enable-tk \
            --disable-schemas-install $SWITCHES
 
-make %{?_smp_mflags} LIBTOOL=/usr/bin/libtool
+make %{?_smp_mflags} LIBTOOL="/usr/bin/libtool --tag=CC"
 
 # one_time_password plugin, included upstream but not built by default
 cd libpurple/plugins/
-make one_time_password.so LIBTOOL=/usr/bin/libtool
+make one_time_password.so LIBTOOL="/usr/bin/libtool --tag=CC"
 cd -
 
 %if %{api_docs}
@@ -593,7 +593,7 @@ find doc/html -empty -delete
 
 %install
 rm -rf $RPM_BUILD_ROOT
-make DESTDIR=$RPM_BUILD_ROOT install LIBTOOL=/usr/bin/libtool
+make DESTDIR=$RPM_BUILD_ROOT install LIBTOOL="/usr/bin/libtool --tag=CC"
 
 install -m 0755 libpurple/plugins/one_time_password.so $RPM_BUILD_ROOT%{_libdir}/purple-2/
 
@@ -796,6 +796,9 @@ rm -rf $RPM_BUILD_ROOT
 %endif
 
 %changelog
+* Fri Nov 04 2016 Jacco Ligthart <jacco@redsleeve.org> 2.10.7-25.el7.redsleeve
+- added "--tag=CC" to the make command due to libtool errors
+
 * Thu Mar 24 2016 Debarshi Ray <rishi@fedoraproject.org> - 2.10.7-26
 - Bump release to be higher than the EPEL build
   Resolves: #1297461
-- 
1.8.3.1

