From 42e606a898ad23f5e5a66e75df8e0ad5448105db Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Fri, 4 Aug 2017 12:14:42 +0200
Subject: [PATCH] added "--tag=CC" to the make command due to libtool errors

---
 SPECS/pidgin.spec | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/SPECS/pidgin.spec b/SPECS/pidgin.spec
index 5610aab..5066bf6 100644
--- a/SPECS/pidgin.spec
+++ b/SPECS/pidgin.spec
@@ -117,7 +117,7 @@
 
 Name:           pidgin
 Version:        2.10.11
-Release:        5%{?dist}
+Release:        5%{?dist}.redsleeve
 License:        GPLv2+ and GPLv2 and MIT
 # GPLv2+ - libpurple, gnt, finch, pidgin, most prpls
 # GPLv2 - silc & novell prpls
@@ -552,11 +552,11 @@ autoreconf --force --install
            --enable-tcl --enable-tk \
            --disable-schemas-install $SWITCHES
 
-make %{?_smp_mflags} LIBTOOL=/usr/bin/libtool
+make %{?_smp_mflags} LIBTOOL="/usr/bin/libtool --tag=CC"
 
 # one_time_password plugin, included upstream but not built by default
 cd libpurple/plugins/
-make one_time_password.so LIBTOOL=/usr/bin/libtool
+make one_time_password.so LIBTOOL="/usr/bin/libtool --tag=CC"
 cd -
 
 %if %{api_docs}
@@ -566,7 +566,7 @@ find doc/html -empty -delete
 
 %install
 rm -rf $RPM_BUILD_ROOT
-make DESTDIR=$RPM_BUILD_ROOT install LIBTOOL=/usr/bin/libtool
+make DESTDIR=$RPM_BUILD_ROOT install LIBTOOL="/usr/bin/libtool --tag=CC"
 
 install -m 0755 libpurple/plugins/one_time_password.so $RPM_BUILD_ROOT%{_libdir}/purple-2/
 
@@ -770,6 +770,9 @@ rm -rf $RPM_BUILD_ROOT
 %endif
 
 %changelog
+* Fri Aug 04 2017 Jacco Ligthart <jacco@redsleeve.org> 2.10.11-5.el7.redsleeve
+- added "--tag=CC" to the make command due to libtool errors
+
 * Fri May 19 2017 Debarshi Ray <rishi@fedoraproject.org> - 2.10.11-5
 - Drop MXit support in RHEL
   Resolves: #1439296
-- 
1.8.3.1

