From a5e82ccb366a0d9fd6fb7a2df66ffe4d9aa17936 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sun, 15 Apr 2018 14:50:20 +0200
Subject: [PATCH] added "--tag=CC" to the make command due to libtool errors

---
 SPECS/pidgin.spec | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/SPECS/pidgin.spec b/SPECS/pidgin.spec
index 6b82d11..01abaf3 100644
--- a/SPECS/pidgin.spec
+++ b/SPECS/pidgin.spec
@@ -110,7 +110,7 @@
 
 Name:           pidgin
 Version:        2.10.11
-Release:        7%{?dist}
+Release:        7%{?dist}.redsleeve
 License:        GPLv2+ and GPLv2 and MIT
 # GPLv2+ - libpurple, gnt, finch, pidgin, most prpls
 # GPLv2 - novell prpls
@@ -545,11 +545,11 @@ autoreconf --force --install
            --enable-tcl --enable-tk \
            --disable-schemas-install $SWITCHES
 
-make %{?_smp_mflags} LIBTOOL=/usr/bin/libtool
+make %{?_smp_mflags} LIBTOOL="/usr/bin/libtool --tag=CC"
 
 # one_time_password plugin, included upstream but not built by default
 cd libpurple/plugins/
-make one_time_password.so LIBTOOL=/usr/bin/libtool
+make one_time_password.so LIBTOOL="/usr/bin/libtool --tag=CC"
 cd -
 
 %if %{api_docs}
@@ -559,7 +559,7 @@ find doc/html -empty -delete
 
 %install
 rm -rf $RPM_BUILD_ROOT
-make DESTDIR=$RPM_BUILD_ROOT install LIBTOOL=/usr/bin/libtool
+make DESTDIR=$RPM_BUILD_ROOT install LIBTOOL="/usr/bin/libtool --tag=CC"
 
 install -m 0755 libpurple/plugins/one_time_password.so $RPM_BUILD_ROOT%{_libdir}/purple-2/
 
@@ -763,6 +763,9 @@ rm -rf $RPM_BUILD_ROOT
 %endif
 
 %changelog
+* Sun Apr 15 2018 Jacco Ligthart <jacco@redsleeve.org> 2.10.11-7.el7.redsleeve
+- added "--tag=CC" to the make command due to libtool errors
+
 * Thu Oct 12 2017 Debarshi Ray <rishi@fedoraproject.org> - 2.10.11-7
 - Drop AIM in RHEL
   Resolves: #1500403
-- 
1.8.3.1

