From efa7a144ca2611743be38b4beb80d6c57fafae00 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 10 Aug 2019 10:19:08 +0200
Subject: [PATCH] added "--tag=CC" to the make command due to libtool errors

---
 SPECS/pidgin.spec | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/SPECS/pidgin.spec b/SPECS/pidgin.spec
index e854332..54f6b8b 100644
--- a/SPECS/pidgin.spec
+++ b/SPECS/pidgin.spec
@@ -110,7 +110,7 @@
 
 Name:           pidgin
 Version:        2.10.11
-Release:        8%{?dist}
+Release:        8%{?dist}.redsleeve
 License:        GPLv2+ and GPLv2 and MIT
 # GPLv2+ - libpurple, gnt, finch, pidgin, most prpls
 # GPLv2 - novell prpls
@@ -553,11 +553,11 @@ autoreconf --force --install
            --enable-tcl --enable-tk \
            --disable-schemas-install $SWITCHES
 
-make %{?_smp_mflags} LIBTOOL=/usr/bin/libtool
+make %{?_smp_mflags} LIBTOOL="/usr/bin/libtool --tag=CC"
 
 # one_time_password plugin, included upstream but not built by default
 cd libpurple/plugins/
-make one_time_password.so LIBTOOL=/usr/bin/libtool
+make one_time_password.so LIBTOOL="/usr/bin/libtool --tag=CC"
 cd -
 
 %if %{api_docs}
@@ -567,7 +567,7 @@ find doc/html -empty -delete
 
 %install
 rm -rf $RPM_BUILD_ROOT
-make DESTDIR=$RPM_BUILD_ROOT install LIBTOOL=/usr/bin/libtool
+make DESTDIR=$RPM_BUILD_ROOT install LIBTOOL="/usr/bin/libtool --tag=CC"
 
 install -m 0755 libpurple/plugins/one_time_password.so $RPM_BUILD_ROOT%{_libdir}/purple-2/
 
@@ -771,6 +771,9 @@ rm -rf $RPM_BUILD_ROOT
 %endif
 
 %changelog
+* Sat Aug 10 2019 Jacco Ligthart <jacco@redsleeve.org> 2.10.11-8.el7.redsleeve
+- added "--tag=CC" to the make command due to libtool errors
+
 * Tue Oct 09 2018 Debarshi Ray <rishi@fedoraproject.org> - 2.10.11-8
 - Rework tray icon blinking
 - Unbreak connections to Freenode
-- 
1.8.3.1

