diff -uNrp sos-3.7.orig/sos/policies/redhat.py sos-3.7/sos/policies/redhat.py
--- sos-3.7.orig/sos/policies/redhat.py	2020-03-18 00:35:41.357473057 +0000
+++ sos-3.7/sos/policies/redhat.py	2020-03-18 00:37:31.569721326 +0000
@@ -31,9 +31,9 @@ OS_RELEASE = "/etc/os-release"
 
 
 class RedHatPolicy(LinuxPolicy):
-    distro = "Red Hat"
-    vendor = "Red Hat"
-    vendor_url = "http://www.redhat.com/"
+    distro = "CentOS"
+    vendor = "CentOS"
+    vendor_url = "http://www,centos.org/"
     _redhat_release = '/etc/redhat-release'
     _tmp_dir = "/var/tmp"
     _rpmq_cmd = 'rpm -qa --queryformat "%{NAME}|%{VERSION}|%{RELEASE}\\n"'
@@ -92,9 +92,9 @@ class RedHatPolicy(LinuxPolicy):
 
     @classmethod
     def check(cls):
-        """This method checks to see if we are running on Red Hat. It must be
+        """This method checks to see if we are running on CentOS. It must be
         overriden by concrete subclasses to return True when running on a
-        Fedora, RHEL or other Red Hat distribution or False otherwise."""
+        Fedora, RHEL or CentOS distribution or False otherwise."""
         return False
 
     def check_usrmove(self, pkgs):
@@ -185,7 +185,7 @@ class RedHatPolicy(LinuxPolicy):
         return self.host_name()
 
 
-# Container environment variables on Red Hat systems.
+# Container environment variables on CentOS systems.
 ENV_CONTAINER = 'container'
 ENV_HOST_SYSROOT = 'HOST'
 
@@ -196,7 +196,7 @@ _cb_plugs = ['abrt', 'block', 'boot', 'd
              'hardware', 'host', 'kernel', 'logs', 'lvm2', 'memory', 'rpm',
              'process', 'systemd', 'yum', 'xfs']
 
-RHEL_RELEASE_STR = "Red Hat Enterprise Linux"
+RHEL_RELEASE_STR = "CentOS Linux"
 
 RHV = "rhv"
 RHV_DESC = "Red Hat Virtualization"
@@ -205,17 +205,17 @@ RHEL = "rhel"
 RHEL_DESC = RHEL_RELEASE_STR
 
 RHOSP = "rhosp"
-RHOSP_DESC = "Red Hat OpenStack Platform"
+RHOSP_DESC = "RDO"
 RHOSP_OPTS = SoSOptions(plugopts=[
                              'process.lsof=off',
                              'networking.ethtool_namespaces=False',
                              'networking.namespaces=200'])
 
 RHOCP = "ocp"
-RHOCP_DESC = "OpenShift Container Platform by Red Hat"
+RHOCP_DESC = "OpenShift"
 
 RH_SATELLITE = "satellite"
-RH_SATELLITE_DESC = "Red Hat Satellite"
+RH_SATELLITE_DESC = "Satellite"
 SAT_OPTS = SoSOptions(verify=True, plugopts=['apache.log=on'])
 
 CB = "cantboot"
@@ -253,9 +253,9 @@ No changes will be made to system config
 
 
 class RHELPolicy(RedHatPolicy):
-    distro = RHEL_RELEASE_STR
-    vendor = "Red Hat"
-    vendor_url = "https://access.redhat.com/support/"
+    distro = "CentOS Linux"
+    vendor = "CentOS"
+    vendor_url = "https://wiki.centos.org/"
     msg = _("""\
 This command will collect diagnostic and configuration \
 information from this %(distro)s system and installed \
@@ -274,7 +274,7 @@ support representative.
     def check(cls):
         """Test to see if the running host is a RHEL installation.
 
-            Checks for the presence of the "Red Hat Enterprise Linux"
+            Checks for the presence of the "CentOS Linux"
             release string at the beginning of the NAME field in the
             `/etc/os-release` file and returns ``True`` if it is
             found, and ``False`` otherwise.
@@ -343,7 +343,7 @@ class CentOsPolicy(RHELPolicy):
 
 ATOMIC = "atomic"
 ATOMIC_RELEASE_STR = "Atomic"
-ATOMIC_DESC = "Red Hat Enterprise Linux Atomic Host"
+ATOMIC_DESC = "CentOS Linux Atomic Host"
 
 atomic_presets = {
     ATOMIC: PresetDefaults(name=ATOMIC, desc=ATOMIC_DESC, note=NOTE_TIME,
@@ -352,7 +352,7 @@ atomic_presets = {
 
 
 class RedHatAtomicPolicy(RHELPolicy):
-    distro = "Red Hat Atomic Host"
+    distro = "CentOS Atomic Host"
     msg = _("""\
 This command will collect diagnostic and configuration \
 information from this %(distro)s system.
