diff -uNrp sos-3.8.orig/sos/policies/redhat.py sos-3.8/sos/policies/redhat.py
--- sos-3.8.orig/sos/policies/redhat.py	2020-05-12 15:45:00.128929276 +0000
+++ sos-3.8/sos/policies/redhat.py	2020-05-12 15:59:16.328778927 +0000
@@ -23,9 +23,9 @@ OS_RELEASE = "/etc/os-release"
 
 
 class RedHatPolicy(LinuxPolicy):
-    distro = "Red Hat"
-    vendor = "Red Hat"
-    vendor_url = "http://www.redhat.com/"
+    distro = "CentOS"
+    vendor = "CentOS"
+    vendor_url = "http://www.centos.org/"
     _redhat_release = '/etc/redhat-release'
     _tmp_dir = "/var/tmp"
     _rpmq_cmd = 'rpm -qa --queryformat "%{NAME}|%{VERSION}|%{RELEASE}\\n"'
@@ -77,9 +77,9 @@ class RedHatPolicy(LinuxPolicy):
 
     @classmethod
     def check(cls):
-        """This method checks to see if we are running on Red Hat. It must be
+        """This method checks to see if we are running on CentOS. It must be
         overriden by concrete subclasses to return True when running on a
-        Fedora, RHEL or other Red Hat distribution or False otherwise."""
+        Fedora, RHEL or CentOS distribution or False otherwise."""
         return False
 
     def check_usrmove(self, pkgs):
@@ -178,26 +178,26 @@ _cb_plugs = ['abrt', 'block', 'boot', 'd
              'hardware', 'host', 'kernel', 'logs', 'lvm2', 'memory', 'rpm',
              'process', 'systemd', 'yum', 'xfs']
 
-RHEL_RELEASE_STR = "Red Hat Enterprise Linux"
+RHEL_RELEASE_STR = "CentOS Linux"
 
 RHV = "rhv"
-RHV_DESC = "Red Hat Virtualization"
+RHV_DESC = "Virtualization"
 
 RHEL = "rhel"
 RHEL_DESC = RHEL_RELEASE_STR
 
 RHOSP = "rhosp"
-RHOSP_DESC = "Red Hat OpenStack Platform"
+RHOSP_DESC = "OpenStack"
 RHOSP_OPTS = SoSOptions(plugopts=[
                              'process.lsof=off',
                              'networking.ethtool_namespaces=False',
                              'networking.namespaces=200'])
 
 RHOCP = "ocp"
-RHOCP_DESC = "OpenShift Container Platform by Red Hat"
+RHOCP_DESC = "RDO"
 
 RH_SATELLITE = "satellite"
-RH_SATELLITE_DESC = "Red Hat Satellite"
+RH_SATELLITE_DESC = "Satellite"
 SAT_OPTS = SoSOptions(verify=True, plugopts=['apache.log=on'])
 
 CB = "cantboot"
@@ -236,9 +236,9 @@ No changes will be made to system config
 
 
 class RHELPolicy(RedHatPolicy):
-    distro = RHEL_RELEASE_STR
-    vendor = "Red Hat"
-    vendor_url = "https://access.redhat.com/support/"
+    distro = "CentOS Linux"
+    vendor = "CentOS"
+    vendor_url = "https://wiki.centos.org/"
     msg = _("""\
 This command will collect diagnostic and configuration \
 information from this %(distro)s system and installed \
@@ -257,7 +257,7 @@ support representative.
     def check(cls):
         """Test to see if the running host is a RHEL installation.
 
-            Checks for the presence of the "Red Hat Enterprise Linux"
+            Checks for the presence of the "CentOS Linux"
             release string at the beginning of the NAME field in the
             `/etc/os-release` file and returns ``True`` if it is
             found, and ``False`` otherwise.
@@ -314,7 +314,7 @@ class CentOsPolicy(RHELPolicy):
 
 ATOMIC = "atomic"
 ATOMIC_RELEASE_STR = "Atomic"
-ATOMIC_DESC = "Red Hat Enterprise Linux Atomic Host"
+ATOMIC_DESC = "CentOS Linux Atomic Host"
 
 atomic_presets = {
     ATOMIC: PresetDefaults(name=ATOMIC, desc=ATOMIC_DESC, note=NOTE_TIME,
@@ -323,7 +323,7 @@ atomic_presets = {
 
 
 class RedHatAtomicPolicy(RHELPolicy):
-    distro = "Red Hat Atomic Host"
+    distro = "CentOS Atomic Host"
     msg = _("""\
 This command will collect diagnostic and configuration \
 information from this %(distro)s system.
