From c746b80e91bbab8c40a0c8cc5607a221b2bed09f Mon Sep 17 00:00:00 2001
From: "Bryn M. Reeves" <bmr@redhat.com>
Date: Fri, 15 Jul 2016 12:02:40 +0100
Subject: [PATCH] [monit] fix add_copy_spec() arguments

Signed-off-by: Bryn M. Reeves <bmr@redhat.com>
---
 sos/plugins/monit.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sos/plugins/monit.py b/sos/plugins/monit.py
index 825f770..9b7d30f 100644
--- a/sos/plugins/monit.py
+++ b/sos/plugins/monit.py
@@ -27,6 +27,7 @@ class Monit(Plugin, RedHatPlugin):
     plugin_name = 'monit'
 
     # Define configuration files
+    # FIXME: direct globs will fail in container environments.
     monit_conf = glob("/etc/monit.d/*")
     monit_conf.append("/etc/monit.conf")
     monit_conf.append("/etc/monitrc")
@@ -38,7 +39,7 @@ class Monit(Plugin, RedHatPlugin):
 
     def setup(self):
         self.add_cmd_output("monit status")
-        self.add_copy_spec([self.monit_log, self.monit_conf])
+        self.add_copy_spec(self.monit_log + self.monit_conf)
 
     def postproc(self):
         # Post process the files included to scrub any
-- 
2.4.11

