From 9b1f9472e7b0b8a993f635c8d4c757b59c46ed0f Mon Sep 17 00:00:00 2001
From: "Bryn M. Reeves" <bmr@redhat.com>
Date: Wed, 27 Mar 2019 21:00:36 +0000
Subject: [PATCH] [sos] bump release

Signed-off-by: Bryn M. Reeves <bmr@redhat.com>
---
 sos.spec        | 2 +-
 sos/__init__.py | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/sos.spec b/sos.spec
index 95249670..68aedcfd 100644
--- a/sos.spec
+++ b/sos.spec
@@ -2,7 +2,7 @@
 
 Summary: A set of tools to gather troubleshooting information from a system
 Name: sos
-Version: 3.6
+Version: 3.7
 Release: 1%{?dist}
 Group: Applications/System
 Source0: http://people.redhat.com/breeves/sos/releases/sos-%{version}.tar.gz
diff --git a/sos/__init__.py b/sos/__init__.py
index c436bd20..dfc7ed5f 100644
--- a/sos/__init__.py
+++ b/sos/__init__.py
@@ -25,7 +25,7 @@ if six.PY3:
 else:
     from ConfigParser import ConfigParser, ParsingError, Error
 
-__version__ = "3.6"
+__version__ = "3.7"
 
 gettext_dir = "/usr/share/locale"
 gettext_app = "sos"
-- 
2.17.2

