--- a/hw/core/Makefile.objs	2018-02-20 06:44:30.029587435 +0100
+++ b/hw/core/Makefile.objs	2018-02-20 06:42:50.434759721 +0100
@@ -16,8 +16,8 @@
 common-obj-$(CONFIG_FITLOADER) += loader-fit.o
 common-obj-$(CONFIG_SOFTMMU) += qdev-properties-system.o
 common-obj-$(CONFIG_PLATFORM_BUS) += platform-bus.o
+common-obj-$(CONFIG_SOFTMMU) += register.o
 # Disabled in Red Hat Enterprise Linux
-# common-obj-$(CONFIG_SOFTMMU) += register.o
 # obj-$(CONFIG_SOFTMMU) += generic-loader.o
 # common-obj-$(CONFIG_SOFTMMU) += or-irq.o
 
