--- a/tests/Makefile.include	2018-02-20 19:41:07.624205352 +0100
+++ b/tests/Makefile.include	2018-02-20 19:48:53.453680881 +0100
@@ -323,13 +323,13 @@
 #gcov-files-sparc64-y += hw/timer/m48t59.c
 check-qtest-sparc64-y += tests/prom-env-test$(EXESUF)
 
-check-qtest-arm-y = tests/tmp105-test$(EXESUF)
-check-qtest-arm-y += tests/ds1338-test$(EXESUF)
-check-qtest-arm-y += tests/m25p80-test$(EXESUF)
+#check-qtest-arm-y = tests/tmp105-test$(EXESUF)
+#check-qtest-arm-y += tests/ds1338-test$(EXESUF)
+#check-qtest-arm-y += tests/m25p80-test$(EXESUF)
 gcov-files-arm-y += hw/misc/tmp105.c
-check-qtest-arm-y += tests/virtio-blk-test$(EXESUF)
+check-qtest-arm-y = tests/virtio-blk-test$(EXESUF)
 gcov-files-arm-y += arm-softmmu/hw/block/virtio-blk.c
-check-qtest-arm-y += tests/test-arm-mptimer$(EXESUF)
+#check-qtest-arm-y += tests/test-arm-mptimer$(EXESUF)
 gcov-files-arm-y += hw/timer/arm_mptimer.c
 
 #check-qtest-aarch64-y = tests/numa-test$(EXESUF)
