From 97ffe8ba32ac08955cdc8712a6a86e931e0e7f67 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Mon, 6 Feb 2017 16:05:44 +0100
Subject: [PATCH] rebrand vendorzone

---
 SPECS/ntp.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/ntp.spec b/SPECS/ntp.spec
index 3a7e62a..41cd5fc 100644
--- a/SPECS/ntp.spec
+++ b/SPECS/ntp.spec
@@ -1,7 +1,7 @@
 Summary: The NTP daemon and utilities
 Name: ntp
 Version: 4.2.6p5
-Release: 25%{?dist}.1
+Release: 25%{?dist}.1.redsleeve
 # primary license (COPYRIGHT) : MIT
 # ElectricFence/ (not used) : GPLv2
 # kernel/sys/ppsclock.h (not used) : BSD with advertising
@@ -264,7 +264,7 @@ This package contains NTP documentation in HTML format.
 # pool.ntp.org vendor zone which will be used in ntp.conf
 %if 0%{!?vendorzone:1}
 %{?fedora: %global vendorzone fedora.}
-%{?rhel: %global vendorzone centos.}
+%{?rhel: %global vendorzone redsleeve.}
 %endif
 
 %prep
@@ -557,6 +557,9 @@ popd
 %{ntpdocdir}/html
 
 %changelog
+* Mon Feb 06 2017 Jacco Ligthart <jacco@redsleeve.org> - 4.2.6p5-25.el7.redsleeve.1
+- rebrand vendorzone
+
 * Sun Feb 05 2017 CentOS Sources <bugs@centos.org> - 4.2.6p5-25.el7.centos.1
 - rebrand vendorzone
 
-- 
1.8.3.1

