From c6784593562f462e80fd998e8f7b5ccf5caf5faf Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Thu, 21 Sep 2017 22:08:46 +0200
Subject: [PATCH] added kvm_target arm

---
 SPECS/qemu-kvm.spec | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/SPECS/qemu-kvm.spec b/SPECS/qemu-kvm.spec
index d7f6d4f..9ee43f9 100644
--- a/SPECS/qemu-kvm.spec
+++ b/SPECS/qemu-kvm.spec
@@ -14,7 +14,7 @@
     %global have_usbredir 0
 %endif
 
-%ifnarch s390 s390x
+%ifnarch s390 s390x %{arm}
     %global have_librdma 1
     %global have_tcmalloc 1
 %endif
@@ -41,6 +41,10 @@
 %ifarch aarch64
     %global kvm_target    aarch64
 %endif
+%ifarch %{arm}
+    %global kvm_target    arm
+%endif
+
 
 #Versions of various parts:
 
@@ -76,13 +80,13 @@ Obsoletes: %1 < %{obsoletes_version}                                      \
 Summary: QEMU is a machine emulator and virtualizer
 Name: %{pkgname}%{?pkgsuffix}
 Version: 1.5.3
-Release: 141%{?dist}.2
+Release: 141%{?dist}.2.redsleeve
 # Epoch because we pushed a qemu-1.0 package. AIUI this can't ever be dropped
 Epoch: 10
 License: GPLv2+ and LGPLv2+ and BSD
 Group: Development/Tools
 URL: http://www.qemu.org/
-ExclusiveArch: x86_64 %{power64} aarch64 s390x
+ExclusiveArch: x86_64 %{power64} aarch64 s390x %{arm}
 Requires: seabios-bin >= 1.7.2.2-5
 Requires: sgabios-bin
 Requires: seavgabios-bin
@@ -5996,6 +6000,10 @@ sh %{_sysconfdir}/sysconfig/modules/kvm.modules &> /dev/null || :
 %{_mandir}/man8/qemu-nbd.8*
 
 %changelog
+* Thu Sep 21 2017 Jacco Ligthart <jacco@redsleeve.org> - 1.5.3-141.el7.2.redsleeve
+- added kvm_target arm
+- do not use rdma-core
+
 * Mon Aug 21 2017 Miroslav Rezanina <mrezanin@redhat.com> - 1.5.3-141.el7_4.2
 - kvm-virtio-net-dynamic-network-offloads-configuration.patch [bz#1482468]
 - kvm-Workaround-rhel6-ctrl_guest_offloads-machine-type-mi.patch [bz#1482468]
-- 
1.8.3.1

