From fae5041e519c113db31cb0d49933e257e6330d5f Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 3 Aug 2019 00:39:05 +0200
Subject: [PATCH] added kvm_target arm

---
 SPECS/qemu-kvm.spec | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/SPECS/qemu-kvm.spec b/SPECS/qemu-kvm.spec
index b7e76d5..073e7b9 100644
--- a/SPECS/qemu-kvm.spec
+++ b/SPECS/qemu-kvm.spec
@@ -14,7 +14,7 @@
     %global have_usbredir 0
 %endif
 
-%ifnarch s390 s390x
+%ifnarch s390 s390x %{arm}
     %global have_librdma 1
     %global have_tcmalloc 1
 %endif
@@ -41,6 +41,10 @@
 %ifarch aarch64
     %global kvm_target    aarch64
 %endif
+%ifarch %{arm}
+    %global kvm_target    arm
+%endif
+
 
 #Versions of various parts:
 
@@ -76,13 +80,13 @@ Obsoletes: %1 < %{obsoletes_version}                                      \
 Summary: QEMU is a machine emulator and virtualizer
 Name: %{pkgname}%{?pkgsuffix}
 Version: 1.5.3
-Release: 160%{?dist}.3
+Release: 160%{?dist}.3.redsleeve
 # Epoch because we pushed a qemu-1.0 package. AIUI this can't ever be dropped
 Epoch: 10
 License: GPLv2 and GPLv2+ and CC-BY
 Group: Development/Tools
 URL: http://www.qemu.org/
-ExclusiveArch: x86_64
+ExclusiveArch: x86_64 %{arm}
 Requires: seabios-bin >= 1.7.2.2-5
 Requires: sgabios-bin
 Requires: seavgabios-bin
@@ -6522,6 +6526,10 @@ sh %{_sysconfdir}/sysconfig/modules/kvm.modules &> /dev/null || :
 %{_mandir}/man8/qemu-nbd.8*
 
 %changelog
+* Fri Aug 02 2019 Jacco Ligthart <jacco@redsleeve.org> - 1.5.3-160.el7.3.redsleeve
+- added kvm_target arm
+- do not use rdma-core
+
 * Tue May 28 2019 Miroslav Rezanina <mrezanin@redhat.com> - 1.5.3-160.el7_6.3
 - kvm-slirp-check-sscanf-result-when-emulating-ident.patch [bz#1669067]
 - kvm-slirp-fix-big-little-endian-conversion-in-ident-prot.patch [bz#1669067]
-- 
1.8.3.1

