From 4500f0c1d5cdb2cdec8af7de2eccfa94506cd3f7 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sun, 29 Jan 2017 17:01:22 +0100
Subject: [PATCH] added kvm_target arm

---
 SPECS/qemu-kvm.spec | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/SPECS/qemu-kvm.spec b/SPECS/qemu-kvm.spec
index 490bffe..6b7f558 100644
--- a/SPECS/qemu-kvm.spec
+++ b/SPECS/qemu-kvm.spec
@@ -41,6 +41,10 @@
 %ifarch aarch64
     %global kvm_target    aarch64
 %endif
+%ifarch %{arm}
+    %global kvm_target    arm
+%endif
+
 
 #Versions of various parts:
 
@@ -76,7 +80,7 @@ Obsoletes: %1 < %{obsoletes_version}                                      \
 Summary: QEMU is a FAST! processor emulator
 Name: %{pkgname}%{?pkgsuffix}
 Version: 1.5.3
-Release: 126%{?dist}.3
+Release: 126%{?dist}.3.redsleeve
 # Epoch because we pushed a qemu-1.0 package. AIUI this can't ever be dropped
 Epoch: 10
 License: GPLv2+ and LGPLv2+ and BSD
@@ -5712,6 +5716,9 @@ sh %{_sysconfdir}/sysconfig/modules/kvm.modules &> /dev/null || :
 %{_mandir}/man8/qemu-nbd.8*
 
 %changelog
+* Sun Jan 29 2017 Jacco Ligthart <jacco@redsleeve.org> - 1.5.3-126.el7.3.redsleeve
+- added kvm_target arm
+
 * Wed Jan 04 2017 Miroslav Rezanina <mrezanin@redhat.com> - 1.5.3-126.el7_3.3
 - kvm-net-check-packet-payload-length.patch [bz#1398217]
 - Resolves: bz#1398217
-- 
1.8.3.1

