From bd709ba2a2aeea2efee132fa2138c58a49af92aa Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Thu, 7 Sep 2017 10:33:40 +0200
Subject: [PATCH] removed the gdb section of the SPEC file

---
 SPECS/java-1.8.0-openjdk.spec | 29 ++++++++++++++++-------------
 1 file changed, 16 insertions(+), 13 deletions(-)

diff --git a/SPECS/java-1.8.0-openjdk.spec b/SPECS/java-1.8.0-openjdk.spec
index 66a76e7..1f54ea1 100644
--- a/SPECS/java-1.8.0-openjdk.spec
+++ b/SPECS/java-1.8.0-openjdk.spec
@@ -792,7 +792,7 @@ Provides: java-%{javaver}-%{origin}-accessibility = %{epoch}:%{version}-%{releas
 
 Name:    java-%{javaver}-%{origin}
 Version: %{javaver}.%{updatever}
-Release: 0.%{buildver}%{?dist}
+Release: 0.%{buildver}%{?dist}.redsleeve
 # java-1.5.0-ibm from jpackage.org set Epoch to 1 for unknown reasons,
 # and this change was brought into RHEL-4.  java-1.5.0-ibm packages
 # also included the epoch in their virtual provides.  This created a
@@ -1539,18 +1539,18 @@ do
 done
 
 # Make sure gdb can do a backtrace based on line numbers on libjvm.so
-gdb -q "$JAVA_HOME/bin/java" <<EOF | tee gdb.out
-handle SIGSEGV pass nostop noprint
-handle SIGILL pass nostop noprint
-set breakpoint pending on
-break javaCalls.cpp:1
-commands 1
-backtrace
-quit
-end
-run -version
-EOF
-grep 'JavaCallWrapper::JavaCallWrapper' gdb.out
+#gdb -q "$JAVA_HOME/bin/java" <<EOF | tee gdb.out
+#handle SIGSEGV pass nostop noprint
+#handle SIGILL pass nostop noprint
+#set breakpoint pending on
+#break javaCalls.cpp:1
+#commands 1
+#backtrace
+#quit
+#end
+#run -version
+#EOF
+#grep 'JavaCallWrapper::JavaCallWrapper' gdb.out
 
 # Check src.zip has all sources. See RHBZ#1130490
 jar -tf $JAVA_HOME/src.zip | grep 'sun.misc.Unsafe'
@@ -1962,6 +1962,9 @@ require "copy_jdk_configs.lua"
 %endif
 
 %changelog
+* Thu Sep 07 2017 Jacco Ligthart <jacco@ligthart.nu> 1:1.8.0.144-0.b01.redsleeve
+- removed the gdb section of the SPEC file
+
 * Tue Aug 15 2017 Andrew Hughes <gnu.andrew@redhat.com> - 1:1.8.0.144-0.b01
 - Update to aarch64-jdk8u144-b01 and aarch64-shenandoah-jdk8u144-b01.
 - Exclude 8175887 from Shenandoah builds as it has been included in that repo.
-- 
1.8.3.1

