--- current/find-provides.ksyms.orig	2017-10-04 15:25:12.448353956 +0200
+++ current/find-provides.ksyms	2017-10-04 15:29:10.086785421 +0200
@@ -1,10 +1,24 @@
-#! /bin/sh
+#! /bin/bash
 
 IFS=$'\n'
 
 for module in $(grep -E '/lib/modules/.+\.ko$'); do
-    nm $module \
-    | sed -r -ne 's:^0*([0-9a-f]+) A __crc_(.+):0x\1 \2:p' \
-    | awk --non-decimal-data '{printf("ksym(%s) = 0x%08x\n", $2, $1)}'
-done \
-| sort -u
+    if [[ -n $(nm $module | sed -r -ne 's:^0*([0-9a-f]+) A __crc_(.+):0x\1 \2:p') ]]; then
+        nm $module \
+        | sed -r -ne 's:^0*([0-9a-f]+) A __crc_(.+):0x\1 \2:p' \
+        | awk --non-decimal-data '{printf("ksym(%s) = 0x%08x\n", $2, $1)}' \
+        | LC_ALL=C sort -u
+    else
+        ELFRODATA=$(readelf -R .rodata $module | awk '/0x/{printf $2$3$4$5}')
+        if [[ -n $(readelf -h $module | grep "little endian") ]]; then
+            RODATA=$(echo $ELFRODATA | sed 's/\(..\)\(..\)\(..\)\(..\)/\4\3\2\1/g')
+        else
+            RODATA=$ELFRODATA
+        fi
+        for sym in $(nm $module | sed -r -ne 's:^0*([0-9a-f]+) R __crc_(.+):0x\1 \2:p'); do
+            echo $sym $RODATA
+        done \
+        | awk --non-decimal-data '{printf("ksym(%s) = 0x%08s\n", $2, substr($3,($1*2)+1,8))}' \
+        | LC_ALL=C sort -u
+    fi
+done
