diff -Nuar redhat-rpm-config-9.1.0_old/kmodtool redhat-rpm-config-9.1.0/kmodtool
--- redhat-rpm-config-9.1.0_old/kmodtool	2013-09-22 10:15:55.903436121 +0800
+++ redhat-rpm-config-9.1.0/kmodtool	2013-09-22 10:18:24.230435836 +0800
@@ -154,8 +154,8 @@
     cat <<EOF
 Provides:         kernel-modules >= ${verrel_dep}${dotvariant}
 Provides:         ${kmod_name}-kmod = %{?epoch:%{epoch}:}%{version}-%{release}
-Requires(post):   /sbin/depmod
-Requires(postun): /sbin/depmod
+Requires(post):   /usr/sbin/depmod
+Requires(postun): /usr/sbin/depmod
 EOF
 
     if [ "yes" != "$nobuildreqs" ]
@@ -183,7 +183,7 @@
 cat <<EOF
 %post          -n kmod-${kmod_name}${dashvariant}
 if [ -e "/boot/System.map-${verrel}${dotvariant}" ]; then
-    /sbin/depmod -aeF "/boot/System.map-${verrel}${dotvariant}" "${verrel}${dotvariant}" > /dev/null || :
+    /usr/sbin/depmod -aeF "/boot/System.map-${verrel}${dotvariant}" "${verrel}${dotvariant}" > /dev/null || :
 fi
 
 modules=( \$(find /lib/modules/${verrel}${dotvariant}/extra/${kmod_name} | grep '\.ko$') )
@@ -201,7 +201,7 @@
 cat <<EOF
 %postun        -n kmod-${kmod_name}${dashvariant}
 if [ -e "/boot/System.map-${verrel}${dotvariant}" ]; then
-    /sbin/depmod -aeF "/boot/System.map-${verrel}${dotvariant}" "${verrel}${dotvariant}" > /dev/null || :
+    /usr/sbin/depmod -aeF "/boot/System.map-${verrel}${dotvariant}" "${verrel}${dotvariant}" > /dev/null || :
 fi
 
 modules=( \$(cat /var/run/rpm-kmod-${kmod_name}${dashvariant}-modules) )
