From 22a17e277b1e93efb8be3a055ad6ab1439aa9e89 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sun, 11 Dec 2016 15:12:20 +0100
Subject: [PATCH] use the redsleeve NTP pool

---
 SPECS/system-config-date.spec | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/SPECS/system-config-date.spec b/SPECS/system-config-date.spec
index c4f51fa..0ae862d 100644
--- a/SPECS/system-config-date.spec
+++ b/SPECS/system-config-date.spec
@@ -26,7 +26,7 @@
 Summary: A graphical interface for modifying system date and time
 Name: system-config-date
 Version: 1.10.6
-Release: 2%{?dist}
+Release: 2%{?dist}.redsleeve
 URL: http://fedorahosted.org/%{name}
 License: GPLv2+
 Group: System Environment/Base
@@ -87,8 +87,8 @@ make \
 %if 0%{?rhel} > 0
     POOL_NTP_ORG_VENDOR=rhel \
 %endif
-%if 0%{?centos_ver} > 0
-    POOL_NTP_ORG_VENDOR=centos \
+%if 0%{?redsleeve_ver} > 0
+    POOL_NTP_ORG_VENDOR=redsleeve \
 %endif
     %{?_smp_mflags}
 
@@ -134,6 +134,9 @@ fi
 #%{python_sitelib}/scdate.dbus-%{version}-py%{python_version}.egg-info
 
 %changelog
+* Tue May 19 2015 Jacco Ligthart <jacco@redsleeve.org> - 1.10.6-2.el7
+- change POOL_NTP_ORG_VENDOR=redsleeve if redsleeve_ver >0
+
 * Thu Jun 19 2014 Johnny Hughes <johnny@centos.org> - 1.10.6-2.el7
 - change POOL_NTP_ORG_VENDOR=centos if centos_ver >0
 
-- 
1.8.3.1

