diff -up mesa-18.0.5/src/glx/drisw_glx.c.dma mesa-18.0.5/src/glx/drisw_glx.c
--- mesa-18.0.5/src/glx/drisw_glx.c.dma	2018-12-07 10:42:42.332659700 +1000
+++ mesa-18.0.5/src/glx/drisw_glx.c	2018-12-07 10:43:08.312349783 +1000
@@ -812,6 +812,8 @@ static int
 check_xshm(Display *dpy)
 {
    int ignore;
+   /* disable shm paths for now */
+   return 0;
 
    return XQueryExtension(dpy, "MIT-SHM", &xshm_opcode, &ignore, &ignore);
 }
