diff -ruN numactl-2.0.9-orig/syscall.c numactl-2.0.9/syscall.c
--- numactl-2.0.9-orig/syscall.c	2013-10-08 23:34:58.000000000 +0200
+++ numactl-2.0.9/syscall.c	2014-12-25 15:56:29.028437246 +0100
@@ -109,6 +109,9 @@
 
 #define __NR_migrate_pages	272
 
+#elif defined (__arm__)
+#define __NR_migrate_pages     380
+
 #elif !defined(DEPS_RUN)
 #error "Add syscalls for your architecture or update kernel headers"
 #endif
