diff -up webkitgtk-2.14.5/Source/WebCore/platform/gtk/PlatformKeyboardEventGtk.cpp.b169028 webkitgtk-2.14.5/Source/WebCore/platform/gtk/PlatformKeyboardEventGtk.cpp
--- webkitgtk-2.14.5/Source/WebCore/platform/gtk/PlatformKeyboardEventGtk.cpp.b169028	2017-03-01 15:46:43.834929604 +0100
+++ webkitgtk-2.14.5/Source/WebCore/platform/gtk/PlatformKeyboardEventGtk.cpp	2017-03-01 15:47:32.663655336 +0100
@@ -578,6 +578,7 @@ PlatformKeyboardEvent::PlatformKeyboardE
     , m_windowsVirtualKeyCode(windowsKeyCodeForGdkKeyCode(event->keyval))
     , m_nativeVirtualKeyCode(event->keyval)
     , m_macCharCode(0)
+    , m_handledByInputMethod(false)
     , m_autoRepeat(false)
     , m_isKeypad(event->keyval >= GDK_KP_Space && event->keyval <= GDK_KP_9)
     , m_isSystemKey(false)
diff -up webkitgtk-2.14.5/Source/WebCore/platform/gtk/PlatformMouseEventGtk.cpp.b169028 webkitgtk-2.14.5/Source/WebCore/platform/gtk/PlatformMouseEventGtk.cpp
--- webkitgtk-2.14.5/Source/WebCore/platform/gtk/PlatformMouseEventGtk.cpp.b169028	2017-03-01 15:46:34.898979963 +0100
+++ webkitgtk-2.14.5/Source/WebCore/platform/gtk/PlatformMouseEventGtk.cpp	2017-03-01 15:47:06.047804427 +0100
@@ -41,6 +41,7 @@ PlatformMouseEvent::PlatformMouseEvent(G
     m_timestamp = event->time;
     m_position = IntPoint((int)event->x, (int)event->y);
     m_globalPosition = IntPoint((int)event->x_root, (int)event->y_root);
+    m_modifierFlags = 0;
 
     m_modifiers = 0;
     if (event->state & GDK_SHIFT_MASK)
@@ -86,6 +87,7 @@ PlatformMouseEvent::PlatformMouseEvent(G
     m_timestamp = motion->time;
     m_position = IntPoint((int)motion->x, (int)motion->y);
     m_globalPosition = IntPoint((int)motion->x_root, (int)motion->y_root);
+    m_modifierFlags = 0;
 
     m_modifiers = 0;
     if (motion->state & GDK_SHIFT_MASK)
