diff -up webkitgtk-2.14.5/Source/WebCore/platform/gtk/PlatformMouseEventGtk.cpp.b169666 webkitgtk-2.14.5/Source/WebCore/platform/gtk/PlatformMouseEventGtk.cpp
--- webkitgtk-2.14.5/Source/WebCore/platform/gtk/PlatformMouseEventGtk.cpp.b169666	2017-03-20 10:51:21.992722970 +0100
+++ webkitgtk-2.14.5/Source/WebCore/platform/gtk/PlatformMouseEventGtk.cpp	2017-03-20 10:51:29.303679688 +0100
@@ -41,6 +41,8 @@ PlatformMouseEvent::PlatformMouseEvent(G
     m_timestamp = event->time;
     m_position = IntPoint((int)event->x, (int)event->y);
     m_globalPosition = IntPoint((int)event->x_root, (int)event->y_root);
+    m_button = NoButton;
+    m_clickCount = 0;
     m_modifierFlags = 0;
 
     m_modifiers = 0;
@@ -87,6 +89,8 @@ PlatformMouseEvent::PlatformMouseEvent(G
     m_timestamp = motion->time;
     m_position = IntPoint((int)motion->x, (int)motion->y);
     m_globalPosition = IntPoint((int)motion->x_root, (int)motion->y_root);
+    m_button = NoButton;
+    m_clickCount = 0;
     m_modifierFlags = 0;
 
     m_modifiers = 0;
