Index: /Source/JavaScriptCore/runtime/Options.cpp
===================================================================
--- /Source/JavaScriptCore/runtime/Options.cpp	(revision 213241)
+++ /Source/JavaScriptCore/runtime/Options.cpp	(revision 213242)
@@ -102,7 +102,12 @@
 static bool parse(const char* string, const char*& value)
 {
-    if (!strlen(string))
-        string = nullptr;
-    value = string;
+    if (!strlen(string)) {
+        value = nullptr;
+        return true;
+    }
+
+    // FIXME <https://webkit.org/b/169057>: This could leak if this option is set more than once.
+    // Given that Options are typically used for testing, this isn't considered to be a problem.
+    value = WTF::fastStrDup(string);
     return true;
 }
@@ -223,12 +228,12 @@
     }
     
-    m_rangeString = rangeString;
-
-    if (*rangeString == '!') {
+    const char* p = rangeString;
+
+    if (*p == '!') {
         invert = true;
-        rangeString++;
-    }
-
-    int scanResult = sscanf(rangeString, " %u:%u", &m_lowLimit, &m_highLimit);
+        p++;
+    }
+
+    int scanResult = sscanf(p, " %u:%u", &m_lowLimit, &m_highLimit);
 
     if (!scanResult || scanResult == EOF) {
@@ -245,4 +250,7 @@
     }
 
+    // FIXME <https://webkit.org/b/169057>: This could leak if this particular option is set more than once.
+    // Given that these options are used for testing, this isn't considered to be problem.
+    m_rangeString = WTF::fastStrDup(rangeString);
     m_state = invert ? Inverted : Normal;
 
