diff -up webkitgtk-2.14.5/Source/WebCore/platform/graphics/texmap/BitmapTexturePool.h.b169665 webkitgtk-2.14.5/Source/WebCore/platform/graphics/texmap/BitmapTexturePool.h
--- webkitgtk-2.14.5/Source/WebCore/platform/graphics/texmap/BitmapTexturePool.h.b169665	2017-02-07 09:05:08.000000000 +0100
+++ webkitgtk-2.14.5/Source/WebCore/platform/graphics/texmap/BitmapTexturePool.h	2017-03-20 10:48:08.356853704 +0100
@@ -59,7 +59,7 @@ private:
         void markIsInUse() { m_lastUsedTime = monotonicallyIncreasingTime(); }
 
         RefPtr<BitmapTexture> m_texture;
-        double m_lastUsedTime;
+        double m_lastUsedTime { 0.0 };
     };
 
     void scheduleReleaseUnusedTextures();
diff -up webkitgtk-2.14.5/Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.h.b169665 webkitgtk-2.14.5/Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.h
--- webkitgtk-2.14.5/Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.h.b169665	2017-02-07 09:05:08.000000000 +0100
+++ webkitgtk-2.14.5/Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.h	2017-03-20 10:48:08.357853698 +0100
@@ -256,7 +256,7 @@ private:
 #endif
     Timer m_animationStartedTimer;
     TextureMapperAnimations m_animations;
-    double m_lastAnimationStartTime;
+    double m_lastAnimationStartTime { 0.0 };
 
     ScrollableArea* m_scrollableArea;
 };
diff -up webkitgtk-2.14.5/Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsState.h.b169665 webkitgtk-2.14.5/Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsState.h
--- webkitgtk-2.14.5/Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsState.h.b169665	2017-03-20 10:48:08.357853698 +0100
+++ webkitgtk-2.14.5/Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsState.h	2017-03-20 10:49:31.564376743 +0100
@@ -131,6 +131,7 @@ struct CoordinatedGraphicsLayerState {
         , replica(InvalidCoordinatedLayerID)
         , mask(InvalidCoordinatedLayerID)
         , imageID(InvalidCoordinatedImageBackingID)
+        , repaintCount(0)
 #if USE(GRAPHICS_SURFACE)
         , platformLayerFrontBuffer(0)
 #endif
diff -up webkitgtk-2.14.5/Source/WebCore/platform/graphics/texmap/coordinated/SurfaceUpdateInfo.h.b169665 webkitgtk-2.14.5/Source/WebCore/platform/graphics/texmap/coordinated/SurfaceUpdateInfo.h
--- webkitgtk-2.14.5/Source/WebCore/platform/graphics/texmap/coordinated/SurfaceUpdateInfo.h.b169665	2017-02-07 09:05:08.000000000 +0100
+++ webkitgtk-2.14.5/Source/WebCore/platform/graphics/texmap/coordinated/SurfaceUpdateInfo.h	2017-03-20 10:48:08.357853698 +0100
@@ -35,7 +35,7 @@ public:
     IntRect updateRect;
 
     // The id of the update atlas including the shareable bitmap containing the updates.
-    uint32_t atlasID;
+    uint32_t atlasID { 0 };
 
     // The offset in the bitmap where the rendered contents are.
     IntPoint surfaceOffset;
