diff -up webkitgtk-2.14.5/Source/JavaScriptCore/runtime/Options.cpp.b169029 webkitgtk-2.14.5/Source/JavaScriptCore/runtime/Options.cpp
--- webkitgtk-2.14.5/Source/JavaScriptCore/runtime/Options.cpp.b169029	2017-03-01 16:41:10.931855353 +0100
+++ webkitgtk-2.14.5/Source/JavaScriptCore/runtime/Options.cpp	2017-03-01 16:40:19.277138520 +0100
@@ -527,6 +527,7 @@ bool Options::setOptions(const char* opt
         p = strstr(p, "=");
         if (!p) {
             dataLogF("'=' not found in option string: %p\n", optionStart);
+            WTF::fastFree(optionsStrCopy);
             return false;
         }
         p++;
@@ -538,6 +539,7 @@ bool Options::setOptions(const char* opt
             p = strstr(p + 1, "\"");
             if (!p) {
                 dataLogF("Missing trailing '\"' in option string: %p\n", optionStart);
+                WTF::fastFree(optionsStrCopy);
                 return false; // End of string not found.
             }
             hasStringValue = true;
@@ -575,6 +577,9 @@ bool Options::setOptions(const char* opt
     }
 
     dumpOptionsIfNeeded();
+
+    WTF::fastFree(optionsStrCopy);
+
     return success;
 }
 
