diff -up webkitgtk-2.14.3/Source/WebCore/Modules/indexeddb/server/SQLiteIDBBackingStore.cpp.prid64 webkitgtk-2.14.3/Source/WebCore/Modules/indexeddb/server/SQLiteIDBBackingStore.cpp
--- webkitgtk-2.14.3/Source/WebCore/Modules/indexeddb/server/SQLiteIDBBackingStore.cpp.prid64	2017-01-17 12:45:51.000000000 +0100
+++ webkitgtk-2.14.3/Source/WebCore/Modules/indexeddb/server/SQLiteIDBBackingStore.cpp	2017-02-09 11:37:57.495158658 +0100
@@ -52,6 +52,7 @@
 #include <runtime/JSGlobalObject.h>
 #include <runtime/StructureInlines.h>
 #include <wtf/NeverDestroyed.h>
+#include <wtf/StdLibExtras.h>
 
 using namespace JSC;
 
diff -up webkitgtk-2.14.3/Source/WTF/wtf/StdLibExtras.h.prid64 webkitgtk-2.14.3/Source/WTF/wtf/StdLibExtras.h
--- webkitgtk-2.14.3/Source/WTF/wtf/StdLibExtras.h.prid64	2017-01-17 12:45:51.000000000 +0100
+++ webkitgtk-2.14.3/Source/WTF/wtf/StdLibExtras.h	2017-02-09 11:51:51.944926864 +0100
@@ -29,6 +29,8 @@
 
 #include <chrono>
 #include <cstring>
+#define __STDC_FORMAT_MACROS
+#include <inttypes.h>
 #include <memory>
 #include <wtf/Assertions.h>
 #include <wtf/CheckedArithmetic.h>
