diff --git a/Source/JavaScriptCore/heap/MarkedBlock.h b/Source/JavaScriptCore/heap/MarkedBlock.h
index 7d14eecfd8d..95ea41c04dd 100644
--- a/Source/JavaScriptCore/heap/MarkedBlock.h
+++ b/Source/JavaScriptCore/heap/MarkedBlock.h
@@ -66,7 +66,12 @@ private:
     friend class Handle;
 public:
     static constexpr size_t atomSize = 16; // bytes
+
+#if CPU(PPC64) || CPU(PPC64LE) || CPU(PPC) || CPU(ARM64) || CPU(UNKNOWN)
+    static constexpr size_t blockSize = 64 * KB;
+#else
     static constexpr size_t blockSize = 16 * KB;
+#endif
     static constexpr size_t blockMask = ~(blockSize - 1); // blockSize must be a power of two.
 
     static constexpr size_t atomsPerBlock = blockSize / atomSize;
