From 24ec9b4487e5732cca5457b959ff99b3f45e149e Mon Sep 17 00:00:00 2001
From: Tomas Popela <tpopela@redhat.com>
Date: Tue, 5 Sep 2017 12:03:20 +0200
Subject: [PATCH] wip


diff --git a/Source/WebCore/platform/URLParser.cpp b/Source/WebCore/platform/URLParser.cpp
index 154512c0527..96feb85fbef 100644
--- a/Source/WebCore/platform/URLParser.cpp
+++ b/Source/WebCore/platform/URLParser.cpp
@@ -1253,6 +1253,7 @@ void URLParser::parse(const CharacterType* input, const unsigned length, const U
                     m_asciiBuffer.clear();
                     state = State::NoScheme;
                     c = beginAfterControlAndSpace;
+                    break;
                 }
                 state = State::Scheme;
             } else
-- 
2.13.5

