--- webkitgtk-2.22.7/Source/WebCore/accessibility/AccessibilityObject.cpp.atk_crash	2019-02-28 11:08:20.000000000 +0100
+++ webkitgtk-2.22.7/Source/WebCore/accessibility/AccessibilityObject.cpp	2019-03-25 14:21:15.340450227 +0100
@@ -70,6 +70,7 @@
 #include "RenderWidget.h"
 #include "RenderedPosition.h"
 #include "RuntimeEnabledFeatures.h"
+#include "ScriptDisallowedScope.h"
 #include "Settings.h"
 #include "TextCheckerClient.h"
 #include "TextCheckingHelper.h"
@@ -1783,7 +1784,10 @@ void AccessibilityObject::updateBackingS
     // Updating the layout may delete this object.
     RefPtr<AccessibilityObject> protectedThis(this);
     if (auto* document = this->document()) {
-        if (!document->view()->layoutContext().isInRenderTreeLayout() && !document->inRenderTreeUpdate() && !document->inStyleRecalc())
+        if (!document->view()->layoutContext().isInRenderTreeLayout()
+                && !document->inRenderTreeUpdate()
+                && !document->inStyleRecalc()
+                && ScriptDisallowedScope::InMainThread::isScriptAllowed())
             document->updateLayoutIgnorePendingStylesheets();
     }
 
