diff -ruN a/Source/JavaScriptCore/CMakeLists.txt b/Source/JavaScriptCore/CMakeLists.txt
--- a/Source/JavaScriptCore/CMakeLists.txt	2018-07-31 07:57:41.000000000 +0200
+++ b/Source/JavaScriptCore/CMakeLists.txt	2018-11-24 08:57:28.944383874 +0100
@@ -116,9 +116,9 @@
     ${LLVM_LIBRARIES}
 )
 
-if (ATOMIC_INT64_REQUIRES_LIBATOMIC)
+#if (ATOMIC_INT64_REQUIRES_LIBATOMIC)
     list(APPEND JavaScriptCore_LIBRARIES atomic)
-endif ()
+#endif ()
 
 set(JavaScriptCore_SCRIPTS_SOURCES_DIR "${JAVASCRIPTCORE_DIR}/Scripts")
 
diff -ruN a/Source/WebKit/CMakeLists.txt b/Source/WebKit/CMakeLists.txt
--- a/Source/WebKit/CMakeLists.txt	2018-06-11 11:28:11.000000000 +0200
+++ b/Source/WebKit/CMakeLists.txt	2018-11-24 08:56:41.194424930 +0100
@@ -778,9 +778,9 @@
     set(JavaScriptCore_SCRIPTS_DIR "${FORWARDING_HEADERS_DIR}/JavaScriptCore/Scripts")
 endif ()
 
-if (ATOMIC_INT64_REQUIRES_LIBATOMIC)
-    list(APPEND WebKit_LIBRARIES atomic)
-endif ()
+#if (ATOMIC_INT64_REQUIRES_LIBATOMIC)
+#    list(APPEND WebKit_LIBRARIES atomic)
+#endif ()
 
 if (UNIX)
     check_function_exists(shm_open SHM_OPEN_EXISTS)
