diff -up webkitgtk-2.16.6/Source/WebKit2/PlatformGTK.cmake.cmake-fix webkitgtk-2.16.6/Source/WebKit2/PlatformGTK.cmake
--- webkitgtk-2.16.6/Source/WebKit2/PlatformGTK.cmake.cmake-fix	2017-08-17 11:22:28.342080871 +0200
+++ webkitgtk-2.16.6/Source/WebKit2/PlatformGTK.cmake	2017-08-17 11:23:03.067950392 +0200
@@ -943,8 +943,6 @@ list(APPEND WebKit2_LIBRARIES
 )
 endif ()
 
-ADD_WHOLE_ARCHIVE_TO_LIBRARIES(WebKit2_LIBRARIES)
-
 set(WebKit2_MARSHAL_LIST ${WEBKIT2_DIR}/UIProcess/API/gtk/webkit2marshal.list)
 add_custom_command(
     OUTPUT ${DERIVED_SOURCES_WEBKIT2GTK_API_DIR}/WebKitMarshal.cpp
