diff -up webkitgtk-2.20.3/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp.covscan_gstreamer webkitgtk-2.20.3/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp
--- webkitgtk-2.20.3/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp.covscan_gstreamer	2018-06-27 10:21:44.715668445 +0200
+++ webkitgtk-2.20.3/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp	2018-06-27 10:22:48.513607713 +0200
@@ -227,9 +227,9 @@ public:
     bool isValid() const { return m_isValid; }
 
 private:
-    GstVideoFrame m_videoFrame;
+    GstVideoFrame m_videoFrame { };
     IntSize m_size;
-    TextureMapperGL::Flags m_flags;
+    TextureMapperGL::Flags m_flags { };
     GLuint m_textureID { 0 };
     bool m_isValid { false };
 };
diff -up webkitgtk-2.20.3/Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.h.covscan_gstreamer webkitgtk-2.20.3/Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.h
--- webkitgtk-2.20.3/Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.h.covscan_gstreamer	2018-02-19 08:45:32.000000000 +0100
+++ webkitgtk-2.20.3/Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.h	2018-06-27 10:21:44.715668445 +0200
@@ -85,7 +85,7 @@ private:
     MediaSourceGStreamer* m_mediaSource;
     ContentType m_type;
     Ref<MediaSourceClientGStreamerMSE> m_client;
-    SourceBufferPrivateClient* m_sourceBufferPrivateClient;
+    SourceBufferPrivateClient* m_sourceBufferPrivateClient { nullptr };
     bool m_isReadyForMoreSamples = true;
     bool m_notifyWhenReadyForMoreSamples = false;
     AtomicString m_trackId;
diff -up webkitgtk-2.20.3/Source/WebCore/platform/graphics/gstreamer/VideoSinkGStreamer.cpp.covscan_gstreamer webkitgtk-2.20.3/Source/WebCore/platform/graphics/gstreamer/VideoSinkGStreamer.cpp
--- webkitgtk-2.20.3/Source/WebCore/platform/graphics/gstreamer/VideoSinkGStreamer.cpp.covscan_gstreamer	2018-02-19 08:45:32.000000000 +0100
+++ webkitgtk-2.20.3/Source/WebCore/platform/graphics/gstreamer/VideoSinkGStreamer.cpp	2018-06-27 10:21:44.715668445 +0200
@@ -135,7 +135,7 @@ struct _WebKitVideoSinkPrivate {
 
     VideoRenderRequestScheduler scheduler;
     GstVideoInfo info;
-    GstCaps* currentCaps;
+    GstCaps* currentCaps { nullptr };
 };
 
 #define webkit_video_sink_parent_class parent_class
