diff -up webkitgtk-2.20.3/Source/JavaScriptCore/runtime/JSString.h.covscan_fixes webkitgtk-2.20.3/Source/JavaScriptCore/runtime/JSString.h
--- webkitgtk-2.20.3/Source/JavaScriptCore/runtime/JSString.h.covscan_fixes	2018-06-13 15:02:38.085399191 +0200
+++ webkitgtk-2.20.3/Source/JavaScriptCore/runtime/JSString.h	2018-06-13 15:02:45.140375189 +0200
@@ -221,7 +221,7 @@ private:
     mutable unsigned m_flags;
 
     // A string is represented either by a String or a rope of fibers.
-    unsigned m_length;
+    unsigned m_length{0};
     mutable String m_value;
 
     friend class LLIntOffsetsExtractor;
diff -up webkitgtk-2.20.3/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp.covscan_fixes webkitgtk-2.20.3/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp
--- webkitgtk-2.20.3/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp.covscan_fixes	2018-06-11 10:16:31.000000000 +0200
+++ webkitgtk-2.20.3/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp	2018-06-13 16:14:56.188378149 +0200
@@ -230,7 +230,7 @@ private:
     GstVideoFrame m_videoFrame;
     IntSize m_size;
     TextureMapperGL::Flags m_flags;
-    GLuint m_textureID;
+    GLuint m_textureID { 0 };
     bool m_isValid { false };
 };
 #endif // USE(GSTREAMER_GL)
