From 3b3a788ba1b42e32b794ceb87ce16bd7403585b3 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Fri, 4 Aug 2017 13:10:06 +0200
Subject: [PATCH] added kvm_target arm

---
 SPECS/qemu-guest-agent.spec | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/SPECS/qemu-guest-agent.spec b/SPECS/qemu-guest-agent.spec
index 1104b1a..0a8e11d 100644
--- a/SPECS/qemu-guest-agent.spec
+++ b/SPECS/qemu-guest-agent.spec
@@ -51,6 +51,10 @@
     %global kvm_target    aarch64
     %global have_fdt     1
 %endif
+%ifarch %{arm}
+    %global kvm_target    arm
+    %global have_fdt     1
+%endif
 
 #Versions of various parts:
 
@@ -59,7 +63,7 @@
 Summary: QEMU guest agent
 Name: qemu-guest-agent
 Version: 2.8.0
-Release: 2%{?dist}
+Release: 2%{?dist}.redsleeve
 # Epoch because we pushed a qemu-1.0 package. AIUI this can't ever be dropped
 Epoch: 10
 License: GPLv2+ and LGPLv2+ and BSD
@@ -256,6 +260,9 @@ install -m 0644  qemu-ga.8 ${RPM_BUILD_ROOT}%{_mandir}/man8/
 
 
 %changelog
+* Fri Aug 04 2017 Jacco Ligthart <jacco@redsleeve.org> - 2.8.0-2.el7.redsleeve
+- added kvm_target arm
+
 * Fri May 19 2017 Miroslav Rezanina <mrezanin@redhat.com> - 2.8.0-2.el7
 - qemuga-Remove-unnecessary-dependencies.patch [bz#1441999]
 - Resolves: bz#1441999
-- 
1.8.3.1

