From fab6ea1f80199ca13bd4b45d69900641231082c8 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Wed, 31 Oct 2018 23:59:48 +0100
Subject: [PATCH] added kvm_target arm

---
 SPECS/qemu-guest-agent.spec | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/SPECS/qemu-guest-agent.spec b/SPECS/qemu-guest-agent.spec
index 297bc69..7b66c58 100644
--- a/SPECS/qemu-guest-agent.spec
+++ b/SPECS/qemu-guest-agent.spec
@@ -48,6 +48,10 @@
     %global kvm_target    aarch64
     %global have_fdt     1
 %endif
+%ifarch %{arm}
+    %global kvm_target    arm
+    %global have_fdt     1
+%endif
 
 #Versions of various parts:
 
@@ -56,7 +60,7 @@
 Summary: QEMU guest agent
 Name: qemu-guest-agent
 Version: 2.12.0
-Release: 2%{?dist}
+Release: 2%{?dist}.redsleeve
 # Epoch because we pushed a qemu-1.0 package. AIUI this can't ever be dropped
 Epoch: 10
 License: GPLv2
@@ -259,6 +263,9 @@ install -m 0644  qemu-ga.8 ${RPM_BUILD_ROOT}%{_mandir}/man8/
 
 
 %changelog
+* Wed Oct 31 2018 Jacco Ligthart <jacco@redsleeve.org> - 2.12.0-2.el7.redsleeve
+- added kvm_target arm
+
 * Tue Jul 24 2018 Miroslav Rezanina <mrezanin@redhat.com> - 2.12.0-2.el7
 - qemuga-qemu-ga-make-get-fsinfo-work-over-pci-bridges.patch [bz#1567041]
 - qemuga-qga-fix-driver-leak-in-guest-get-fsinfo.patch [bz#1567041]
-- 
1.8.3.1

