From 97583e33840fc46bc09fcbe9d5decf3f933c0cdd Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sun, 5 Apr 2020 13:37:26 +0200
Subject: [PATCH] undo most of the CentOS work to get an unbranded version.

---
 SPECS/libreport.spec | 123 ++++++++++++++++++++++++++-------------------------
 1 file changed, 63 insertions(+), 60 deletions(-)

diff --git a/SPECS/libreport.spec b/SPECS/libreport.spec
index 0588d39..be871ee 100644
--- a/SPECS/libreport.spec
+++ b/SPECS/libreport.spec
@@ -7,7 +7,7 @@
 Summary: Generic library for reporting various problems
 Name: libreport
 Version: 2.1.11
-Release: 53%{?dist}
+Release: 53%{?dist}.redsleeve
 License: GPLv2+
 Group: System Environment/Libraries
 URL: https://fedorahosted.org/abrt/
@@ -311,21 +311,21 @@ Patch273: 0273-lib-problem_data-Add-a-null-guard-for-g_strsplit-cal.patch
 Patch274: 0274-Update-translations.patch
 # git format-patch 2.1.11-52.el7 -N --start-number=275 --topo-order
 
-Patch1000: 1000-bugzilla-port-to-Problem-Format-API.patch
-Patch1001: 1001-lib-created-a-new-lib-file-for-reporters.patch
+#Patch1000: 1000-bugzilla-port-to-Problem-Format-API.patch
+#Patch1001: 1001-lib-created-a-new-lib-file-for-reporters.patch
 #Patch1002: 1002-spec-changed-spec-file-to-work-with-last-commit.patch
-Patch1003: 1003-ureport-set-url-to-public-faf-server.patch
-Patch1004: 1004-conf-changed-URL-for-sending-uReport.patch
-Patch1005: 1005-reporter-mantisbt-first-version-of-the-reporter-mant.patch
+#Patch1003: 1003-ureport-set-url-to-public-faf-server.patch
+#Patch1004: 1004-conf-changed-URL-for-sending-uReport.patch
+#Patch1005: 1005-reporter-mantisbt-first-version-of-the-reporter-mant.patch
 #Patch1006: 1006-spec-changed-spec-file-to-work-with-reporter-mantisb.patch
-Patch1007: 1007-reporter-mantisbt-change-default-formating-file-for-.patch
+#Patch1007: 1007-reporter-mantisbt-change-default-formating-file-for-.patch
 #Patch1008: 1008-spec-change-spec-file-to-work-with-last-commit.patch
-Patch1009: 1009-reporter-mantisbt-adds-man-pages-for-reporter-mantis.patch
-Patch1010: 1010-move-problem_report-to-plugins.patch
+#Patch1009: 1009-reporter-mantisbt-adds-man-pages-for-reporter-mantis.patch
+#Patch1010: 1010-move-problem_report-to-plugins.patch
 #Patch1011: 1011-spec-change-related-to-moving-problem_report-to-plug.patch
-Patch1012: 1012-reporter-mantisbt-add-event-for-reporting-AVCs.patch
+#Patch1012: 1012-reporter-mantisbt-add-event-for-reporting-AVCs.patch
 #Patch1013: 1013-spec-add-files-related-to-reporting-AVCs-by-reporter.patch
-Patch1014: 1014-event-disable-report_RHTSupport-event-and-change-URL.patch
+#Patch1014: 1014-event-disable-report_RHTSupport-event-and-change-URL.patch
 
 # git is need for '%%autosetup -S git' which automatically applies all the
 # patches above. Please, be aware that the patches must be generated
@@ -515,24 +515,24 @@ Uploads micro-report to abrt server
 %description plugin-bugzilla
 Plugin to report bugs into the bugzilla.
 
-%package plugin-mantisbt
-Summary: %{name}'s mantisbt plugin
-Group: System Environment/Libraries
-Requires: %{name} = %{version}-%{release}
-Requires: libreport-web = %{version}-%{release}
+#%package plugin-mantisbt
+#Summary: %{name}'s mantisbt plugin
+#Group: System Environment/Libraries
+#Requires: %{name} = %{version}-%{release}
+#Requires: libreport-web = %{version}-%{release}
 
-%description plugin-mantisbt
-Plugin to report bugs into the mantisbt.
+#%description plugin-mantisbt
+#Plugin to report bugs into the mantisbt.
 
-%package centos
-Summary: %{name}'s CentOS Bug Tracker workflow
-Group: System Environment/Libraries
-Requires: %{name} = %{version}-%{release}
-Requires: libreport-web = %{version}-%{release}
-Requires: libreport-plugin-mantisbt = %{version}-%{release}
+#%package centos
+#Summary: %{name}'s CentOS Bug Tracker workflow
+#Group: System Environment/Libraries
+#Requires: %{name} = %{version}-%{release}
+#Requires: libreport-web = %{version}-%{release}
+#Requires: libreport-plugin-mantisbt = %{version}-%{release}
 
-%description centos
-Workflows to report issues into the CentOS Bug Tracker.
+#%description centos
+#Workflows to report issues into the CentOS Bug Tracker.
 
 %package plugin-rhtsupport
 Summary: %{name}'s RHTSupport plugin
@@ -779,7 +779,7 @@ gtk-update-icon-cache %{_datadir}/icons/hicolor &>/dev/null || :
 %{_includedir}/libreport/dump_dir.h
 %{_includedir}/libreport/event_config.h
 %{_includedir}/libreport/problem_data.h
-#%{_includedir}/libreport/problem_report.h
+%{_includedir}/libreport/problem_report.h
 %{_includedir}/libreport/report.h
 %{_includedir}/libreport/run_event.h
 %{_includedir}/libreport/file_obj.h
@@ -787,7 +787,7 @@ gtk-update-icon-cache %{_datadir}/icons/hicolor &>/dev/null || :
 %{_includedir}/libreport/workflow.h
 %{_includedir}/libreport/ureport.h
 %{_includedir}/libreport/global_configuration.h
-%{_includedir}/libreport/reporters.h
+#%{_includedir}/libreport/reporters.h
 # Private api headers:
 %{_includedir}/libreport/internal_abrt_dbus.h
 %{_includedir}/libreport/internal_libreport.h
@@ -908,39 +908,39 @@ gtk-update-icon-cache %{_datadir}/icons/hicolor &>/dev/null || :
 %{_mandir}/man5/bugzilla_format_kernel.conf.5.*
 %{_bindir}/reporter-bugzilla
 
-%files plugin-mantisbt
-%defattr(-,root,root,-)
-%config(noreplace) %{_sysconfdir}/libreport/plugins/mantisbt.conf
-%{_datadir}/%{name}/conf.d/plugins/mantisbt.conf
-%config(noreplace) %{_sysconfdir}/libreport/plugins/mantisbt_format.conf
-%config(noreplace) %{_sysconfdir}/libreport/plugins/mantisbt_formatdup.conf
-%config(noreplace) %{_sysconfdir}/libreport/plugins/mantisbt_format_analyzer_libreport.conf
-%config(noreplace) %{_sysconfdir}/libreport/plugins/mantisbt_formatdup_analyzer_libreport.conf
-%{_bindir}/reporter-mantisbt
-%{_mandir}/man1/reporter-mantisbt.1.gz
-%{_mandir}/man5/mantisbt.conf.5.*
-%{_mandir}/man5/mantisbt_format.conf.5.*
-%{_mandir}/man5/mantisbt_formatdup.conf.5.*
-%{_mandir}/man5/mantisbt_format_analyzer_libreport.conf.5.*
-%{_mandir}/man5/mantisbt_formatdup_analyzer_libreport.conf.5.*
-
-%files centos
-%{_datadir}/%{name}/workflows/workflow_CentOSCCpp.xml
-%{_datadir}/%{name}/workflows/workflow_CentOSKerneloops.xml
-%{_datadir}/%{name}/workflows/workflow_CentOSPython.xml
-%{_datadir}/%{name}/workflows/workflow_CentOSPython3.xml
-%{_datadir}/%{name}/workflows/workflow_CentOSVmcore.xml
-%{_datadir}/%{name}/workflows/workflow_CentOSXorg.xml
-%{_datadir}/%{name}/workflows/workflow_CentOSLibreport.xml
-%{_datadir}/%{name}/workflows/workflow_CentOSJava.xml
-%config(noreplace) %{_sysconfdir}/libreport/workflows.d/report_centos.conf
-%{_mandir}/man5/report_centos.conf.5.*
-%{_datadir}/%{name}/events/report_CentOSBugTracker.xml
-%config(noreplace) %{_sysconfdir}/libreport/events/report_CentOSBugTracker.conf
-%{_mandir}/man5/report_CentOSBugTracker.conf.5.*
+#%files plugin-mantisbt
+#%defattr(-,root,root,-)
+#%config(noreplace) %{_sysconfdir}/libreport/plugins/mantisbt.conf
+#%{_datadir}/%{name}/conf.d/plugins/mantisbt.conf
+#%config(noreplace) %{_sysconfdir}/libreport/plugins/mantisbt_format.conf
+#%config(noreplace) %{_sysconfdir}/libreport/plugins/mantisbt_formatdup.conf
+#%config(noreplace) %{_sysconfdir}/libreport/plugins/mantisbt_format_analyzer_libreport.conf
+#%config(noreplace) %{_sysconfdir}/libreport/plugins/mantisbt_formatdup_analyzer_libreport.conf
+#%{_bindir}/reporter-mantisbt
+#%{_mandir}/man1/reporter-mantisbt.1.gz
+#%{_mandir}/man5/mantisbt.conf.5.*
+#%{_mandir}/man5/mantisbt_format.conf.5.*
+#%{_mandir}/man5/mantisbt_formatdup.conf.5.*
+#%{_mandir}/man5/mantisbt_format_analyzer_libreport.conf.5.*
+#%{_mandir}/man5/mantisbt_formatdup_analyzer_libreport.conf.5.*
+
+#%files centos
+#%{_datadir}/%{name}/workflows/workflow_CentOSCCpp.xml
+#%{_datadir}/%{name}/workflows/workflow_CentOSKerneloops.xml
+#%{_datadir}/%{name}/workflows/workflow_CentOSPython.xml
+#%{_datadir}/%{name}/workflows/workflow_CentOSPython3.xml
+#%{_datadir}/%{name}/workflows/workflow_CentOSVmcore.xml
+#%{_datadir}/%{name}/workflows/workflow_CentOSXorg.xml
+#%{_datadir}/%{name}/workflows/workflow_CentOSLibreport.xml
+#%{_datadir}/%{name}/workflows/workflow_CentOSJava.xml
+#%config(noreplace) %{_sysconfdir}/libreport/workflows.d/report_centos.conf
+#%{_mandir}/man5/report_centos.conf.5.*
+#%{_datadir}/%{name}/events/report_CentOSBugTracker.xml
+#%config(noreplace) %{_sysconfdir}/libreport/events/report_CentOSBugTracker.conf
+#%{_mandir}/man5/report_CentOSBugTracker.conf.5.*
 # report_CentOSBugTracker events are shipped by libreport package
-%config(noreplace) %{_sysconfdir}/libreport/events.d/centos_report_event.conf
-%{_mandir}/man5/centos_report_event.conf.5.gz
+#%config(noreplace) %{_sysconfdir}/libreport/events.d/centos_report_event.conf
+#%{_mandir}/man5/centos_report_event.conf.5.gz
 
 %files plugin-rhtsupport
 %defattr(-,root,root,-)
@@ -1053,6 +1053,9 @@ gtk-update-icon-cache %{_datadir}/icons/hicolor &>/dev/null || :
 
 
 %changelog
+* Sun Apr 05 2020 Jacco Ligthart <jacco@redsleeve.org> = 2.1.11-53.redsleeve
+- undo most of the CentOS work to get an unbranded version.
+
 * Mon Nov 11 2019 Ernestas Kulik <ekulik@redhat.com> - 2.1.11-53
 - Rebuild due to translations not being merged
 
-- 
1.8.3.1

