From 6ca5affc2f65fc31c906f1b11769ba0a58b3569c Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Thu, 15 Oct 2020 01:35:38 +0200
Subject: [PATCH] disabled two more tests, as they are known to break on armv5

---
 SPECS/python3.spec | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/SPECS/python3.spec b/SPECS/python3.spec
index 8ad452e..f8fc55b 100644
--- a/SPECS/python3.spec
+++ b/SPECS/python3.spec
@@ -14,7 +14,7 @@ URL: https://www.python.org/
 #  WARNING  When rebasing to a new Python version,
 #           remember to update the python3-docs package as well
 Version: %{pybasever}.8
-Release: 17%{?dist}
+Release: 17%{?dist}.redsleeve
 License: Python
 
 
@@ -1165,6 +1165,8 @@ CheckPython() {
     -x test_distutils \
     -x test_bdist_rpm \
     %ifarch %{arm}
+    -x test_asyncio \
+    -x test_float \
     -x test_gdb \
     %endif 
     %ifarch %{mips64}
@@ -1652,6 +1654,9 @@ CheckPython optimized
 # ======================================================
 
 %changelog
+* Wed Oct 14 2020 Jacco Ligthart <jacco@redsleeve.org> - 3.6.8-17.redsleeve
+- disabled two more tests, as they are known to break on armv5
+
 * Wed May 06 2020 Charalampos Stratakis <cstratak@redhat.com> - 3.6.8-17
 - Overhaul python's FIPS mode support
 Resolves: rhbz#1788459
-- 
1.8.3.1

