From b822cb2d0153ca94c2e3c759a132c9be5f4e2e71 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Fri, 20 Nov 2020 00:13:02 +0100
Subject: [PATCH] disabled two more tests, as they are known to break on armv5

---
 SPECS/python3.spec | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/SPECS/python3.spec b/SPECS/python3.spec
index 09546fb..5cb390b 100644
--- a/SPECS/python3.spec
+++ b/SPECS/python3.spec
@@ -14,7 +14,7 @@ URL: https://www.python.org/
 #  WARNING  When rebasing to a new Python version,
 #           remember to update the python3-docs package as well
 Version: %{pybasever}.8
-Release: 18%{?dist}
+Release: 18%{?dist}.redsleeve
 License: Python
 
 
@@ -1187,6 +1187,8 @@ CheckPython() {
     -x test_distutils \
     -x test_bdist_rpm \
     %ifarch %{arm}
+    -x test_asyncio \
+    -x test_float \
     -x test_gdb \
     %endif 
     %ifarch %{mips64}
@@ -1674,6 +1676,9 @@ CheckPython optimized
 # ======================================================
 
 %changelog
+* Thu Nov 19 2020 Jacco Ligthart <jacco@redsleeve.org> - 3.6.8-18.redsleeve
+- disabled two more tests, as they are known to break on armv5
+
 * Fri Jul 31 2020 Charalampos Stratakis <cstratak@redhat.com> - 3.6.8-18
 - Avoid infinite loop when reading specially crafted TAR files (CVE-2019-20907)
 Resolves: rhbz#1856481
-- 
1.8.3.1

