From 3c98f80e04121cf1877061ce2b2dee5457bc9fb8 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sun, 5 Apr 2020 00:21:18 +0200
Subject: [PATCH] disabled two more tests, as they are known to break on armv5

---
 SPECS/python3.spec | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/SPECS/python3.spec b/SPECS/python3.spec
index 532c0bf..1cc2b62 100644
--- a/SPECS/python3.spec
+++ b/SPECS/python3.spec
@@ -14,7 +14,7 @@ URL: https://www.python.org/
 #  WARNING  When rebasing to a new Python version,
 #           remember to update the python3-docs package as well
 Version: %{pybasever}.8
-Release: 13%{?dist}
+Release: 13%{?dist}.redsleeve
 License: Python
 
 
@@ -1151,6 +1151,8 @@ CheckPython() {
     -x test_distutils \
     -x test_bdist_rpm \
     %ifarch %{arm}
+    -x test_asyncio \
+    -x test_float \
     -x test_gdb \
     %endif 
     %ifarch %{mips64}
@@ -1644,6 +1646,9 @@ CheckPython optimized
 # ======================================================
 
 %changelog
+* Sat Apr 04 2020 Jacco Ligthart <jacco@redsleeve.org> - 3.6.8-13.redsleeve
+- disabled two more tests, as they are known to break on armv5
+
 * Thu Sep 26 2019 Charalampos Stratakis <cstratak@redhat.com> - 3.6.8-13
 - Security fix for CVE-2019-16056
 Resolves: rhbz#1750774
-- 
1.8.3.1

