From e1edf9813ab30443920a6651029d534ea4ff7ed1 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Fri, 16 Aug 2019 15:51:03 +0200
Subject: [PATCH] removed requirements to proprietary CentOS end RHEL libreport

---
 SPECS/abrt.spec | 47 +++++++++++++++++++++++++----------------------
 1 file changed, 25 insertions(+), 22 deletions(-)

diff --git a/SPECS/abrt.spec b/SPECS/abrt.spec
index cbd649c..079b876 100644
--- a/SPECS/abrt.spec
+++ b/SPECS/abrt.spec
@@ -33,7 +33,7 @@
 Summary: Automatic bug detection and reporting tool
 Name: abrt
 Version: 2.1.11
-Release: 55%{?dist}
+Release: 55%{?dist}.redsleeve
 License: GPLv2+
 Group: Applications/System
 URL: https://abrt.readthedocs.org/
@@ -402,7 +402,7 @@ Patch322: 0322-hooks-ccpp-Honor-CreateCoreBacktrace.patch
 # autogen.sh is need to regenerate all the Makefile files
 Patch1006: 1000-Add-autogen.sh.patch
 Patch1000: 1000-event-don-t-run-the-reporter-bugzilla-h-on-RHEL-and-.patch
-Patch1002: 1002-plugin-set-URL-to-retrace-server.patch
+#Patch1002: 1002-plugin-set-URL-to-retrace-server.patch
 Patch1004: 1004-turn-sosreport-off.patch
 Patch1005: 1005-cli-list-revert-patch-7966e5737e8d3af43b1ecdd6a82323.patch
 
@@ -452,9 +452,9 @@ Requires: python-dbus
 Requires: dmidecode
 %endif
 Requires: libreport-plugin-ureport >= %{libreport_ver}
-%if 0%{?rhel}
-Requires: libreport-plugin-rhtsupport
-%endif
+#%if 0%{?rhel}
+#Requires: libreport-plugin-rhtsupport
+#%endif
 
 # we used to have abrt-plugin-bodhi, but we have removed it
 # and we want allow users to update abrt without necessity to
@@ -637,14 +637,14 @@ Requires: abrt-addon-ccpp
 Requires: abrt-addon-python
 Requires: abrt-addon-xorg
 %if 0%{?rhel}
-%if 0%{?centos_ver}
-Requires: libreport-centos >= %{libreport_ver}
-Requires: libreport-plugin-mantisbt >= %{libreport_ver}
-%else
-Requires: libreport-rhel >= %{libreport_ver}
-Requires: libreport-plugin-rhtsupport >= %{libreport_ver}
-%endif
-%else
+#%if 0%{?centos_ver}
+#Requires: libreport-centos >= %{libreport_ver}
+#Requires: libreport-plugin-mantisbt >= %{libreport_ver}
+#%else
+#Requires: libreport-rhel >= %{libreport_ver}
+#Requires: libreport-plugin-rhtsupport >= %{libreport_ver}
+#%endif
+#%else
 Requires: abrt-retrace-client
 Requires: libreport-plugin-bugzilla >= %{libreport_ver}
 Requires: libreport-plugin-logger >= %{libreport_ver}
@@ -676,19 +676,19 @@ Requires: elfutils
 Requires: abrt-gui
 Requires: gnome-abrt
 %if 0%{?rhel}
-%if 0%{?centos_ver}
-Requires: libreport-centos >= %{libreport_ver}
-Requires: libreport-plugin-mantisbt >= %{libreport_ver}
-%else
-Requires: libreport-rhel >= %{libreport_ver}
-Requires: libreport-plugin-rhtsupport >= %{libreport_ver}
-%endif
-%else
+#%if 0%{?centos_ver}
+#Requires: libreport-centos >= %{libreport_ver}
+#Requires: libreport-plugin-mantisbt >= %{libreport_ver}
+#%else
+#Requires: libreport-rhel >= %{libreport_ver}
+#Requires: libreport-plugin-rhtsupport >= %{libreport_ver}
+#%endif
+#%else
 Requires: abrt-retrace-client
 Requires: libreport-plugin-bugzilla >= %{libreport_ver}
 Requires: libreport-plugin-logger >= %{libreport_ver}
 Requires: libreport-plugin-ureport >= %{libreport_ver}
-Requires: libreport-fedora >= %{libreport_ver}
+#Requires: libreport-fedora >= %{libreport_ver}
 %endif
 #Requires: abrt-plugin-firefox
 Provides: bug-buddy = 2.28.0
@@ -1242,6 +1242,9 @@ gtk-update-icon-cache %{_datadir}/icons/hicolor &>/dev/null || :
 %config(noreplace) %{_sysconfdir}/profile.d/abrt-console-notification.sh
 
 %changelog
+* Fri Aug 16 2019 Jacco Ligthart <jacco@redsleeve.org> - 2.1.11-55.el7.redsleeve
+- removed requirements to proprietary CentOS end RHEL libreport packages
+
 * Tue Aug 06 2019 CentOS Sources <bugs@centos.org> - 2.1.11-55.el7.centos
 - Drop RHTS hint
 -  Change by David Mansfield <david@orthanc.cobite.com>
-- 
1.8.3.1

