commit 58ba3b61d276423cef1e1f52bd73c9706cc073fc
Author: Andrew Hughes <gnu.andrew@redhat.com>
Date:   Sat Oct 15 05:37:36 2022 +0100

    Backport e10231248fc100f9dfa08468ac897f60b843857f

diff --git a/src/jdk.localedata/share/classes/sun/util/cldr/resources/common/bcp47/timezone.xml b/src/jdk.localedata/share/classes/sun/util/cldr/resources/common/bcp47/timezone.xml
index 35f8fc86efa..a9b635b4fa4 100644
--- a/src/jdk.localedata/share/classes/sun/util/cldr/resources/common/bcp47/timezone.xml
+++ b/src/jdk.localedata/share/classes/sun/util/cldr/resources/common/bcp47/timezone.xml
@@ -392,7 +392,7 @@ For terms of use, see http://www.unicode.org/copyright.html
             <type name="tvfun" description="Funafuti, Tuvalu" alias="Pacific/Funafuti"/>
             <type name="twtpe" description="Taipei, Taiwan" alias="Asia/Taipei ROC"/>
             <type name="tzdar" description="Dar es Salaam, Tanzania" alias="Africa/Dar_es_Salaam"/>
-            <type name="uaiev" description="Kiev, Ukraine" alias="Europe/Kiev"/>
+            <type name="uaiev" description="Kyiv, Ukraine" alias="Europe/Kiev Europe/Kyiv"/>
             <type name="uaozh" description="Zaporizhia (Zaporozhye), Ukraine" alias="Europe/Zaporozhye"/>
             <type name="uasip" description="Simferopol, Ukraine" alias="Europe/Simferopol"/>
             <type name="uauzh" description="Uzhhorod (Uzhgorod), Ukraine" alias="Europe/Uzhgorod"/>
diff --git a/test/jdk/sun/util/resources/cldr/TimeZoneNamesTest.java b/test/jdk/sun/util/resources/cldr/TimeZoneNamesTest.java
index c2b5672646c..53903d19d3d 100644
--- a/test/jdk/sun/util/resources/cldr/TimeZoneNamesTest.java
+++ b/test/jdk/sun/util/resources/cldr/TimeZoneNamesTest.java
@@ -23,7 +23,7 @@
 
  /*
  * @test
- * @bug 8181157 8202537 8234347 8236548
+ * @bug 8181157 8202537 8234347 8236548 8293834
  * @modules jdk.localedata
  * @summary Checks CLDR time zone names are generated correctly at runtime
  * @run testng/othervm -Djava.locale.providers=CLDR TimeZoneNamesTest
@@ -102,6 +102,24 @@ public class TimeZoneNamesTest {
                                                     "UTC+04:00",
                                                     "heure : Astrakhan",
                                                     "UTC+04:00"},
+            {"Europe/Kyiv",             Locale.US, "Eastern European Standard Time",
+                                                    "GMT+02:00",
+                                                    "Eastern European Summer Time",
+                                                    "GMT+03:00",
+                                                    "Eastern European Time",
+                                                    "GMT+02:00"},
+            {"Europe/Kyiv",             Locale.FRANCE, "heure normale d\u2019Europe de l\u2019Est",
+                                                    "UTC+02:00",
+                                                    "heure d\u2019\u00e9t\u00e9 d\u2019Europe de l\u2019Est",
+                                                    "UTC+03:00",
+                                                    "heure d\u2019Europe de l\u2019Est",
+                                                    "UTC+02:00"},
+            {"Europe/Kyiv",             Locale.GERMANY, "Osteurop\u00e4ische Normalzeit",
+                                                    "OEZ",
+                                                    "Osteurop\u00e4ische Sommerzeit",
+                                                    "OESZ",
+                                                    "Osteurop\u00e4ische Zeit",
+                                                    "OEZ"},
             {"Europe/Saratov",          Locale.US, "Saratov Standard Time",
                                                     "GMT+04:00",
                                                     "Saratov Daylight Time",
