diff -r 1ddf9a99e4ad src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp
--- a/src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp	Wed Aug 22 21:50:12 2018 +0200
+++ b/src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp	Thu Sep 13 13:51:53 2018 +0100
@@ -5745,11 +5745,13 @@
     }
 
     if (vmIntrinsics::is_intrinsic_available(vmIntrinsics::_dsin)) {
-      StubRoutines::_dsin = generate_dsin_dcos(/* isCos = */ false);
+      // disabled pending fix and retest of generated code
+      // StubRoutines::_dsin = generate_dsin_dcos(/* isCos = */ false);
     }
 
     if (vmIntrinsics::is_intrinsic_available(vmIntrinsics::_dcos)) {
-      StubRoutines::_dcos = generate_dsin_dcos(/* isCos = */ true);
+      // disabled pending fix and retest of generated code
+      // StubRoutines::_dcos = generate_dsin_dcos(/* isCos = */ true);
     }
   }
