From 48fcea92c8a75189a8331ce3d2ac250242880126 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sun, 15 Apr 2018 15:37:39 +0200
Subject: [PATCH] rebrand SPEC file

---
 SPECS/redhat-rpm-config.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/redhat-rpm-config.spec b/SPECS/redhat-rpm-config.spec
index 0fcf8d1..5715c76 100644
--- a/SPECS/redhat-rpm-config.spec
+++ b/SPECS/redhat-rpm-config.spec
@@ -1,7 +1,7 @@
 Summary: CentOS specific rpm configuration files
 Name: redhat-rpm-config
 Version: 9.1.0
-Release: 80%{?dist}
+Release: 80%{?dist}.redsleeve
 # No version specified.
 License: GPL+
 Group: Development/System
@@ -120,7 +120,7 @@ Requires: zip
 Provides: system-rpm-config = %{version}-%{release}
 
 %description
-CentOS specific rpm configuration files.
+RedSleeve specific rpm configuration files.
 
 %prep
 %setup -q
@@ -199,6 +199,9 @@ rm -rf ${RPM_BUILD_ROOT}
 %{_sysconfdir}/rpm/*
 
 %changelog
+* Sun Apr 15 2018 Jacco Ligthart <jacco@redsleeve.org> - 9.1.0-80.el7.redsleeve
+- rebrand SPEC file
+
 * Tue Apr 10 2018 CentOS Sources <bugs@centos.org> - 9.1.0-80.el7.centos
 - update check_rhl function in dist.sh
 
-- 
1.8.3.1

