From 1f629b3d2435b4872b2903fd15b9b8ad048e646f Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Thu, 8 Aug 2019 00:21:24 +0200
Subject: [PATCH] rebrand SPEC file

---
 SPECS/redhat-rpm-config.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/redhat-rpm-config.spec b/SPECS/redhat-rpm-config.spec
index d474892..8bd07fd 100644
--- a/SPECS/redhat-rpm-config.spec
+++ b/SPECS/redhat-rpm-config.spec
@@ -1,7 +1,7 @@
 Summary: CentOS specific rpm configuration files
 Name: redhat-rpm-config
 Version: 9.1.0
-Release: 88%{?dist}
+Release: 88%{?dist}.redsleeve
 # No version specified.
 License: GPL+
 Group: Development/System
@@ -129,7 +129,7 @@ Requires: zip
 Provides: system-rpm-config = %{version}-%{release}
 
 %description
-CentOS specific rpm configuration files.
+RedSleeve specific rpm configuration files.
 
 %prep
 %setup -q
@@ -212,6 +212,9 @@ rm -rf ${RPM_BUILD_ROOT}
 %{_sysconfdir}/rpm/*
 
 %changelog
+* Wed Aug 07 2019 Jacco Ligthart <jacco@redsleeve.org> - 9.1.0-88.el7.redsleeve
+- rebrand SPEC file
+
 * Tue Aug 06 2019 CentOS Sources <bugs@centos.org> - 9.1.0-88.el7.centos
 - update check_rhl function in dist.sh
 
-- 
1.8.3.1

