From 60bb6b1e2567c579aa4fd80db6cbbc448d7ca4a3 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Fri, 4 Aug 2017 13:30:30 +0200
Subject: [PATCH] rebrand SPEC file

---
 SPECS/redhat-rpm-config.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/redhat-rpm-config.spec b/SPECS/redhat-rpm-config.spec
index 1f887ab..b9f65b2 100644
--- a/SPECS/redhat-rpm-config.spec
+++ b/SPECS/redhat-rpm-config.spec
@@ -1,7 +1,7 @@
 Summary: CentOS specific rpm configuration files
 Name: redhat-rpm-config
 Version: 9.1.0
-Release: 76%{?dist}
+Release: 76%{?dist}.redsleeve
 # No version specified.
 License: GPL+
 Group: Development/System
@@ -110,7 +110,7 @@ Requires: zip
 Provides: system-rpm-config = %{version}-%{release}
 
 %description
-CentOS specific rpm configuration files.
+RedSleeve specific rpm configuration files.
 
 %prep
 %setup -q
@@ -185,6 +185,9 @@ rm -rf ${RPM_BUILD_ROOT}
 %{_sysconfdir}/rpm/*
 
 %changelog
+* Fri Aug 04 2017 Jacco Ligthart <jacco@redsleeve.org> - 9.1.0-76.el7.redsleeve
+- rebrand SPEC file
+
 * Mon Jul 31 2017 CentOS Sources <bugs@centos.org> - 9.1.0-76.el7.centos
 - update check_rhl function in dist.sh
 
-- 
1.8.3.1

