From 9a4470fdd4472b33ad3e80fe71874297a88443ac Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Wed, 4 Dec 2019 22:31:56 +0100
Subject: [PATCH] disabled tcmalloc for arm

---
 SPECS/389-ds-base.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/389-ds-base.spec b/SPECS/389-ds-base.spec
index 59c8cef..1d116f7 100644
--- a/SPECS/389-ds-base.spec
+++ b/SPECS/389-ds-base.spec
@@ -19,7 +19,7 @@
 %global use_tcmalloc 0
 %global variant base-asan
 %else
-%if %{_arch} != "s390x" && %{_arch} != "s390"
+%ifnarch s390 s390x %{arm}
 %global use_tcmalloc 1
 %else
 %global use_tcmalloc 0
@@ -39,7 +39,7 @@
 Summary:          389 Directory Server (%{variant})
 Name:             389-ds-base
 Version:          1.3.9.1
-Release:          %{?relprefix}12%{?prerel}%{?dist}
+Release:          %{?relprefix}12%{?prerel}%{?dist}.redsleeve
 License:          GPLv3+
 URL:              https://www.port389.org/
 Group:            System Environment/Daemons
@@ -539,6 +539,9 @@ fi
 %{_sysconfdir}/%{pkgname}/dirsrvtests
 
 %changelog
+* Wed Dec 04 2019 Jacco Ligthart <jacco@redsleeve.org> - 1.3.9.1-12.redsleeve
+- disabled tcmalloc for arm
+
 * Fri Nov 1 2019 Mark Reynolds <mreynolds@redhat.com> - 1.3.9.1-12
 - Bump version to 1.3.9.1-12
 - Resolves: Bug 1767622 - CleanAllRUV task limit not enforced
-- 
1.8.3.1

