From 324a3a90ddb9e08e34e5ec78d8f0d056424a53ad Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Fri, 1 Dec 2017 15:09:09 +0100
Subject: [PATCH] disabled tcmalloc for arm

---
 SPECS/389-ds-base.spec | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/SPECS/389-ds-base.spec b/SPECS/389-ds-base.spec
index ed028d0..8b538b5 100644
--- a/SPECS/389-ds-base.spec
+++ b/SPECS/389-ds-base.spec
@@ -11,7 +11,8 @@
 %global use_Socket6 0
 %global use_nunc_stans 1
 
-%if %{_arch} != "s390x" && %{_arch} != "s390"
+#%if %{_arch} != "s390x" && %{_arch} != "s390"
+%ifnarch s390 s390x %{arm}
 %global use_tcmalloc 1
 %else
 %global use_tcmalloc 0
@@ -30,7 +31,7 @@
 Summary:          389 Directory Server (base)
 Name:             389-ds-base
 Version:          1.3.6.1
-Release:          %{?relprefix}24%{?prerel}%{?dist}
+Release:          %{?relprefix}24%{?prerel}%{?dist}.redsleeve
 License:          GPLv3+
 URL:              https://www.port389.org/
 Group:            System Environment/Daemons
@@ -600,6 +601,9 @@ fi
 %{_sysconfdir}/%{pkgname}/dirsrvtests
 
 %changelog
+* Fri Dec 01 2017 Jacco Ligthart <jacco@redsleeve.org> - 1.3.6.1-24.redsleeve
+- disabled tcmalloc for arm
+
 * Fri Nov 10 2017 Mark Reynolds <mreynolds@redhat.com> - 1.3.6.1-24
 - Bump version to 1.3.6.1-24
 - Resolves: Bug 1508978 - replicated MODRDN fails breaking replication
-- 
1.8.3.1

