From 19b9150f2decd141b3e58995820f3391f6758871 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 21 Mar 2020 21:46:21 +0100
Subject: [PATCH] disabled tcmalloc for arm

---
 SPECS/389-ds-base.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/389-ds-base.spec b/SPECS/389-ds-base.spec
index ac72c06..fdb1f08 100644
--- a/SPECS/389-ds-base.spec
+++ b/SPECS/389-ds-base.spec
@@ -19,7 +19,7 @@
 %global use_tcmalloc 0
 %global variant base-asan
 %else
-%if %{_arch} != "s390x" && %{_arch} != "s390"
+%ifnarch s390 s390x %{arm}
 %global use_tcmalloc 1
 %else
 %global use_tcmalloc 0
@@ -39,7 +39,7 @@
 Summary:          389 Directory Server (%{variant})
 Name:             389-ds-base
 Version:          1.3.9.1
-Release:          %{?relprefix}13%{?prerel}%{?dist}
+Release:          %{?relprefix}13%{?prerel}%{?dist}.redsleeve
 License:          GPLv3+
 URL:              https://www.port389.org/
 Group:            System Environment/Daemons
@@ -546,6 +546,9 @@ fi
 %{_sysconfdir}/%{pkgname}/dirsrvtests
 
 %changelog
+* Sat Mar 21 2020 Jacco Ligthart <jacco@redsleeve.org> - 1.3.9.1-13.redsleeve
+- disabled tcmalloc for arm
+
 * Wed Feb 12 2020 Mark Reynolds <mreynolds@redhat.com> - 1.3.9.1-13
 - Bump version to 1.3.9.1-13
 - Resolves: Bug 1801693 - ns-slapd is crashing while restarting ipactl
-- 
1.8.3.1

