From 52fd3d440f8b17dd7cb93e0b2cca4eadbcc5f15b Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sun, 15 Apr 2018 14:35:29 +0200
Subject: [PATCH] SPEC adjusted to build on armv5

---
 SPECS/mesa.spec | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/SPECS/mesa.spec b/SPECS/mesa.spec
index e676192..11da7c0 100644
--- a/SPECS/mesa.spec
+++ b/SPECS/mesa.spec
@@ -7,7 +7,7 @@
 %define with_vdpau 1
 %define with_wayland 1
 
-%ifnarch ppc
+%ifnarch ppc %{arm}
 %define with_radeonsi 1
 %endif
 
@@ -18,7 +18,7 @@
 # S390 doesn't have video cards, but we need swrast for xserver's GLX
 # llvm (and thus llvmpipe) doesn't actually work on ppc32 or s390
 
-%ifnarch s390 ppc
+%ifnarch s390 ppc %{arm}
 %define with_llvm 1
 %endif
 
@@ -61,7 +61,7 @@
 Summary: Mesa graphics libraries
 Name: mesa
 Version: 17.2.3
-Release: 8.%{gitdate}%{?dist}
+Release: 8.%{gitdate}%{?dist}.redsleeve
 License: MIT
 Group: System Environment/Libraries
 URL: http://www.mesa3d.org
@@ -119,7 +119,8 @@ BuildRequires: python-mako
 BuildRequires: gettext
 %if 0%{?with_llvm}
 %if 0%{?with_private_llvm}
-BuildRequires: llvm-private-devel >= 5.0
+#BuildRequires: llvm-private-devel >= 5.0
+BuildRequires: mesa-private-llvm-devel
 %else
 BuildRequires: llvm-devel >= 3.0
 %endif
@@ -540,8 +541,8 @@ rm -rf $RPM_BUILD_ROOT
 # exist on s390x where swrast is llvmpipe, but does exist on s390 where
 # swrast is classic mesa.  this seems like a bug?  in that it probably
 # means the gallium drivers are linking dricore statically?  fixme.
-%{_libdir}/dri/swrast_dri.so
 %if 0%{?with_llvm}
+%{_libdir}/dri/swrast_dri.so
 %{_libdir}/dri/kms_swrast_dri.so
 %endif
 
@@ -670,6 +671,9 @@ rm -rf $RPM_BUILD_ROOT
 %endif
 
 %changelog
+* Sun Apr 15 2018 Jacco Ligthart <jacco@redsleeve.org> - 17.2.3-8.20171019.redsleeve
+- small changes to the spec to make it build on armv5
+
 * Mon Jan 15 2018 Dave Airlie <airlied@redhat.com> - 17.2.3-8.20171019
 - Add missing Intel CFL ids.
 
-- 
1.8.3.1

