From 882eb2e2dd7060c0a774182fef874a211b1a0d67 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Fri, 20 Nov 2020 00:38:03 +0100
Subject: [PATCH] SPEC adjusted to build on armv5

---
 SPECS/mesa.spec | 22 +++++++++++++---------
 1 file changed, 13 insertions(+), 9 deletions(-)

diff --git a/SPECS/mesa.spec b/SPECS/mesa.spec
index 6c18e42..f0e69cb 100644
--- a/SPECS/mesa.spec
+++ b/SPECS/mesa.spec
@@ -7,7 +7,7 @@
 %define with_vdpau 1
 %define with_wayland 1
 
-%ifnarch ppc
+%ifnarch ppc %{arm}
 %define with_radeonsi 1
 %endif
 
@@ -61,7 +61,7 @@
 Summary: Mesa graphics libraries
 Name: mesa
 Version: 18.3.4
-Release: 12%{?dist}
+Release: 12%{?dist}.redsleeve
 License: MIT
 Group: System Environment/Libraries
 URL: http://www.mesa3d.org
@@ -133,7 +133,8 @@ BuildRequires: python-mako
 BuildRequires: gettext
 %if 0%{?with_llvm}
 %if 0%{?with_private_llvm}
-BuildRequires: llvm-private-devel >= 6.0
+#BuildRequires: llvm-private-devel >= 6.0
+BuildRequires: mesa-private-llvm-devel
 %else
 BuildRequires: llvm-devel >= 3.0
 %endif
@@ -380,8 +381,8 @@ grep -q ^/ src/gallium/auxiliary/vl/vl_decoder.c && exit 1
 %patch48 -p1 -b .i965-bufmgr-Handle-NULL-bufmgr-in-brw_bufmgr_get_for
 
 %if 0%{with_private_llvm}
-sed -i 's/\[llvm-config\]/\[llvm-private-config-%{__isa_bits}\]/g' configure.ac
-sed -i 's/`$LLVM_CONFIG --version`/$LLVM_VERSION_MAJOR.$LLVM_VERSION_MINOR-rhel/' configure.ac
+sed -i 's/\[llvm-config\]/\[mesa-private-llvm-config-%{__isa_bits}\]/g' configure.ac
+sed -i 's/`$LLVM_CONFIG --version`/$LLVM_VERSION_MAJOR.$LLVM_VERSION_MINOR-mesa/' configure.ac
 %endif
 
 # need to use libdrm_nouveau2 on F17
@@ -433,7 +434,7 @@ export CXXFLAGS="$RPM_OPT_FLAGS -fno-rtti -fno-exceptions"
     --enable-dri \
 %if %{with_hardware}
     %{?with_vmware:--enable-xa} \
-    --with-gallium-drivers=%{?with_vmware:svga,}%{?with_radeonsi:radeonsi,}%{?with_llvm:swrast,r600,r300,}%{?with_freedreno:freedreno,}nouveau,virgl \
+    --with-gallium-drivers=%{?with_vmware:svga,}%{?with_radeonsi:radeonsi,}%{?with_llvm:swrast,r300,}%{?with_freedreno:freedreno,}nouveau,virgl \
 %else
     --with-gallium-drivers=%{?with_llvm:swrast} \
 %endif
@@ -554,7 +555,7 @@ rm -rf $RPM_BUILD_ROOT
 %endif
 %if 0%{?with_llvm}
 %{_libdir}/dri/r300_dri.so
-%{_libdir}/dri/r600_dri.so
+#%{_libdir}/dri/r600_dri.so
 %if 0%{?with_radeonsi}
 %{_libdir}/dri/radeonsi_dri.so
 %endif
@@ -588,8 +589,8 @@ rm -rf $RPM_BUILD_ROOT
 %defattr(-,root,root,-)
 %{_libdir}/vdpau/libvdpau_nouveau.so.1*
 %if 0%{?with_llvm}
-%{_libdir}/vdpau/libvdpau_r600.so.1*
-%{_libdir}/vdpau/libvdpau_radeonsi.so.1*
+#%{_libdir}/vdpau/libvdpau_r600.so.1*
+#%{_libdir}/vdpau/libvdpau_radeonsi.so.1*
 %endif
 %endif
 %endif
@@ -694,6 +695,9 @@ rm -rf $RPM_BUILD_ROOT
 %endif
 
 %changelog
+* Thu Nov 19 2020 Jacco Ligthart <jacco@redsleeve.org> - 18.3.4-12.redsleeve
+- small changes to the spec to make it build on armv5
+
 * Tue Oct 27 2020 Michel Dänzer <mdaenzer@redhat.com> - 18.3.4-12
 - Fix for defect reported by Coverity/clang (#1803811)
 
-- 
1.8.3.1

