From f3e4a1dac4b6b62d5c5ff7899c74ca7a75a93b57 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Wed, 31 Oct 2018 23:30:05 +0100
Subject: [PATCH] added arm to exclusive archs

---
 SPECS/corosync.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/corosync.spec b/SPECS/corosync.spec
index 1ce46b5..6d59acf 100644
--- a/SPECS/corosync.spec
+++ b/SPECS/corosync.spec
@@ -24,7 +24,7 @@
 Name: corosync
 Summary: The Corosync Cluster Engine and Application Programming Interfaces
 Version: 2.4.3
-Release: 4%{?gitver}%{?dist}
+Release: 4%{?gitver}%{?dist}.redsleeve
 License: BSD
 Group: System Environment/Base
 URL: http://corosync.github.io/corosync/
@@ -35,7 +35,7 @@ Patch1: bz1536219-2-logging-Close-before-and-open-blackbox-after-fork.patch
 Patch2: bz1560468-1-totemcrypto-Check-length-of-the-packet.patch
 
 %if 0%{?rhel}
-ExclusiveArch: i686 x86_64 s390x ppc64le aarch64
+ExclusiveArch: i686 x86_64 s390x ppc64le aarch64 %{arm}
 %endif
 
 # Runtime bits
@@ -524,6 +524,9 @@ fi
 %endif
 
 %changelog
+* Wed Oct 31 2018 Jacco Ligthart <jacco@redsleeve.org> 2.4.3-4.redsleeve
+- added arm to exclusive archs
+
 * Tue Apr 17 2018 Jan Friesse <jfriesse@redhat.com> 2.4.3-4
 - Resolves: rhbz#1422598
 
-- 
1.8.3.1

