From 87a3728a9a12d25c68e304cb7384d795e3769e0c Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Fri, 4 Aug 2017 10:25:42 +0200
Subject: [PATCH] added arm to exclusive archs

---
 SPECS/corosync.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/corosync.spec b/SPECS/corosync.spec
index aa7d3b1..7d4cf85 100644
--- a/SPECS/corosync.spec
+++ b/SPECS/corosync.spec
@@ -23,7 +23,7 @@
 Name: corosync
 Summary: The Corosync Cluster Engine and Application Programming Interfaces
 Version: 2.4.0
-Release: 9%{?gitver}%{?dist}
+Release: 9%{?gitver}%{?dist}.redsleeve
 License: BSD
 Group: System Environment/Base
 URL: http://corosync.github.io/corosync/
@@ -53,7 +53,7 @@ Patch20: bz1434534-2-logconfig-Do-not-overwrite-logger_subsys-priority.patch
 Patch21: bz1445001-1-Main-Call-mlockall-after-fork.patch
 
 %if 0%{?rhel}
-ExclusiveArch: i686 x86_64 s390x ppc64le
+ExclusiveArch: i686 x86_64 s390x ppc64le %{arm}
 %endif
 
 # Runtime bits
@@ -555,6 +555,9 @@ fi
 %endif
 
 %changelog
+* Fri Aug 04 2017 Jacco Ligthart <jacco@redsleeve.org> 2.4.0-9.redsleeve
+- added arm to exclusive archs
+
 * Wed Apr 26 2017 Jan Friesse <jfriesse@redhat.com> 2.4.0-9
 - Resolves: rhbz#1445001
 
-- 
1.8.3.1

