Index: modules/mappers/mod_rewrite.c
===================================================================
--- a/modules/mappers/mod_rewrite.c	(revision 1499025)
+++ b/modules/mappers/mod_rewrite.c	(revision 1528556)
@@ -589,6 +589,18 @@
             return 7;
         }
         break;
+
+    case 'w':
+    case 'W':
+        if (!strncasecmp(uri, "s://", 4)) {        /* ws://     */
+            *sqs = 1;
+            return 5;
+        }
+        else if (!strncasecmp(uri, "ss://", 5)) {  /* wss://    */
+            *sqs = 1;
+            return 6;
+        }
+        break;
     }
 
     return 0;
