From 99bdbf9fae479abf149336640cefe0313e118f3c Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sun, 5 Apr 2020 13:44:04 +0200
Subject: [PATCH] Roll in Branding Change in the SPEC

---
 SPECS/plymouth.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/plymouth.spec b/SPECS/plymouth.spec
index b0270da..084a820 100644
--- a/SPECS/plymouth.spec
+++ b/SPECS/plymouth.spec
@@ -7,7 +7,7 @@
 Summary: Graphical Boot Animation and Logger
 Name: plymouth
 Version: 0.8.9
-Release: 0.33.20140113%{?dist}
+Release: 0.33.20140113%{?dist}.redsleeve
 License: GPLv2+
 Group: System Environment/Base
 Source0: http://freedesktop.org/software/plymouth/releases/%{name}-%{version}.tar.bz2
@@ -241,7 +241,7 @@ Provides: plymouth(system-theme) = %{version}-%{release}
 
 %description theme-charge
 This package contains the "charge" boot splash theme for
-Plymouth. It is the default theme for CentOS Linux.
+Plymouth. It is the default theme for RedSleeve Linux.
 
 %package plugin-script
 Summary: Plymouth "script" plugin
@@ -501,6 +501,9 @@ fi
 %defattr(-, root, root)
 
 %changelog
+* Sun Apr 05 2020 Jacco Ligthart <jacco@redsleeve.org> 0.8.9-0.33.20140113.el7.redsleeve
+- Roll in Branding Change in the SPEC
+
 * Tue Mar 31 2020 CentOS Sources <bugs@centos.org> - 0.8.9-0.33.20140113.el7.centos
 - Roll in Branding Change in the SPEC
 
-- 
1.8.3.1

