From d77693e2f552576d8e8c1af6a5de9684101fcc3f Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sun, 15 Apr 2018 14:52:47 +0200
Subject: [PATCH] Roll in Branding Change in the SPEC

---
 SPECS/plymouth.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/plymouth.spec b/SPECS/plymouth.spec
index 4519c49..ae06c26 100644
--- a/SPECS/plymouth.spec
+++ b/SPECS/plymouth.spec
@@ -7,7 +7,7 @@
 Summary: Graphical Boot Animation and Logger
 Name: plymouth
 Version: 0.8.9
-Release: 0.31.20140113%{?dist}
+Release: 0.31.20140113%{?dist}.redsleeve
 License: GPLv2+
 Group: System Environment/Base
 Source0: http://freedesktop.org/software/plymouth/releases/%{name}-%{version}.tar.bz2
@@ -238,7 +238,7 @@ Provides: plymouth(system-theme) = %{version}-%{release}
 
 %description theme-charge
 This package contains the "charge" boot splash theme for
-Plymouth. It is the default theme for CentOS Linux.
+Plymouth. It is the default theme for RedSleeve Linux.
 
 %package plugin-script
 Summary: Plymouth "script" plugin
@@ -498,6 +498,9 @@ fi
 %defattr(-, root, root)
 
 %changelog
+* Sun Apr 15 2018 Jacco Ligthart <jacco@redsleeve.org> 0.8.9-0.31.20140113.el7.redsleeve
+- Roll in Branding Change in the SPEC
+
 * Tue Apr 10 2018 CentOS Sources <bugs@centos.org> - 0.8.9-0.31.20140113.el7.centos
 - Roll in Branding Change in the SPEC
 
-- 
1.8.3.1

