From 31417ac9d78bbf24301b5a8da917a2f9517880a1 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Fri, 4 Aug 2017 13:41:45 +0200
Subject: [PATCH] disabled jit for arm

---
 SPECS/webkitgtk3.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/webkitgtk3.spec b/SPECS/webkitgtk3.spec
index 2c9b331..6be5123 100644
--- a/SPECS/webkitgtk3.spec
+++ b/SPECS/webkitgtk3.spec
@@ -14,7 +14,7 @@
 
 Name:           webkitgtk3
 Version:        2.4.11
-Release:        2%{?dist}
+Release:        2%{?dist}.redsleeve
 Summary:        GTK+ Web content engine library
 
 Group:          Development/Libraries
@@ -118,7 +118,7 @@ chmod 644 Source/WebCore/html/canvas/CanvasRenderingContext2D.cpp
 %configure                                      \
         --with-gtk=3.0                          \
         --disable-webkit2                       \
-%ifarch s390 s390x ppc %{power64} aarch64
+%ifarch s390 s390x ppc %{power64} aarch64 %{arm}
         --disable-jit                           \
 %else
         --enable-jit                            \
@@ -197,6 +197,9 @@ find $RPM_BUILD_ROOT%{_libdir} -name "*.la" -delete
 %{_datadir}/gtk-doc/html/webkitdomgtk
 
 %changelog
+* Thu Jun 29 2017 Jacco Ligthart <jacco@redsleeve.org> - 2.4.11-2.redsleeve
+- disabled jit for arm
+
 * Mon Feb 27 2017 Tomas Popela <tpopela@redhat.com> - 2.4.11-2
 - Don't build WebKit2 as it's build in webkitgtk4 package
 - Resolves: rhbz#1383614
-- 
1.8.3.1

